// livepeer/funcs/generateImageToImage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateImageToImage = generateImageToImage;
const files_js_1 = require("../lib/files.js");
const M = __importStar(require("../lib/matchers.js"));
const schemas_js_1 = require("../lib/schemas.js");
const security_js_1 = require("../lib/security.js");
const url_js_1 = require("../lib/url.js");
const components = __importStar(require("../models/components/index.js"));
const errors = __importStar(require("../models/errors/index.js"));
const operations = __importStar(require("../models/operations/index.js"));
const blobs_js_1 = require("../types/blobs.js");
const streams_js_1 = require("../types/streams.js");
/**
 * Image To Image
 *
 * @remarks
 * Apply image transformations to a provided image.
 */
async function generateImageToImage(client, request, options) {
    var _a;
    const parsed = (0, schemas_js_1.safeParse)(request, (value) => components.BodyGenImageToImage$outboundSchema.parse(value), "Input validation failed");
    if (!parsed.ok) {
        return parsed;
    }
    const payload = parsed.value;
    const body = new FormData();
    if ((0, blobs_js_1.isBlobLike)(payload.image)) {
        body.append("image", payload.image);
    }
    else if ((0, streams_js_1.isReadableStream)(payload.image.content)) {
        const buffer = await (0, files_js_1.readableStreamToArrayBuffer)(payload.image.content);
        const blob = new Blob([buffer], { type: "application/octet-stream" });
        body.append("image", blob);
    }
    else {
        body.append("image", new Blob([payload.image.content], { type: "application/octet-stream" }), payload.image.fileName);
    }
    body.append("prompt", payload.prompt);
    if (payload.guidance_scale !== undefined) {
        body.append("guidance_scale", String(payload.guidance_scale));
    }
    if (payload.image_guidance_scale !== undefined) {
        body.append("image_guidance_scale", String(payload.image_guidance_scale));
    }
    if (payload.loras !== undefined) {
        body.append("loras", payload.loras);
    }
    if (payload.model_id !== undefined) {
        body.append("model_id", payload.model_id);
    }
    if (payload.negative_prompt !== undefined) {
        body.append("negative_prompt", payload.negative_prompt);
    }
    if (payload.num_images_per_prompt !== undefined) {
        body.append("num_images_per_prompt", String(payload.num_images_per_prompt));
    }
    if (payload.num_inference_steps !== undefined) {
        body.append("num_inference_steps", String(payload.num_inference_steps));
    }
    if (payload.safety_check !== undefined) {
        body.append("safety_check", String(payload.safety_check));
    }
    if (payload.seed !== undefined) {
        body.append("seed", String(payload.seed));
    }
    if (payload.strength !== undefined) {
        body.append("strength", String(payload.strength));
    }
    const path = (0, url_js_1.pathToFunc)("/api/generate/image-to-image")();
    const headers = new Headers({
        Accept: "application/json",
    });
    const secConfig = await (0, security_js_1.extractSecurity)(client._options.apiKey);
    const securityInput = secConfig == null ? {} : { apiKey: secConfig };
    const requestSecurity = (0, security_js_1.resolveGlobalSecurity)(securityInput);
    const context = {
        operationID: "genImageToImage",
        oAuth2Scopes: [],
        resolvedSecurity: requestSecurity,
        securitySource: client._options.apiKey,
        retryConfig: (options === null || options === void 0 ? void 0 : options.retries)
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: (options === null || options === void 0 ? void 0 : options.retryCodes) || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "POST",
        path: path,
        headers: headers,
        body: body,
        timeoutMs: (options === null || options === void 0 ? void 0 : options.timeoutMs) || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return requestRes;
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: ["400", "401", "422", "4XX", "500", "5XX"],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return doResult;
    }
    const response = doResult.value;
    const responseFields = {
        ContentType: (_a = response.headers.get("content-type")) !== null && _a !== void 0 ? _a : "application/octet-stream",
        StatusCode: response.status,
        RawResponse: response,
        Headers: {},
    };
    const [result] = await M.match(M.json(200, operations.GenImageToImageResponse$inboundSchema, {
        key: "ImageResponse",
    }), M.jsonErr(400, errors.GenImageToImageResponseBody$inboundSchema), M.jsonErr(401, errors.GenImageToImageGenerateResponseBody$inboundSchema), M.jsonErr(422, errors.GenImageToImageGenerateResponseResponseBody$inboundSchema), M.jsonErr(500, errors.GenImageToImageGenerateResponse500ResponseBody$inboundSchema), M.fail(["4XX", "5XX"]), M.json("default", operations.GenImageToImageResponse$inboundSchema, {
        key: "studio-api-error",
    }))(response, { extraFields: responseFields });
    if (!result.ok) {
        return result;
    }
    return result;
}
