// livepeer/funcs/generateLlm.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateLlm = generateLlm;
const encodings_js_1 = require("../lib/encodings.js");
const M = __importStar(require("../lib/matchers.js"));
const schemas_js_1 = require("../lib/schemas.js");
const security_js_1 = require("../lib/security.js");
const url_js_1 = require("../lib/url.js");
const components = __importStar(require("../models/components/index.js"));
const errors = __importStar(require("../models/errors/index.js"));
const operations = __importStar(require("../models/operations/index.js"));
/**
 * LLM
 *
 * @remarks
 * Generate text using a language model.
 */
async function generateLlm(client, request, options) {
    var _a;
    const parsed = (0, schemas_js_1.safeParse)(request, (value) => components.BodyGenLLM$outboundSchema.parse(value), "Input validation failed");
    if (!parsed.ok) {
        return parsed;
    }
    const payload = parsed.value;
    const body = Object.entries(payload || {}).map(([k, v]) => {
        return (0, encodings_js_1.encodeBodyForm)(k, v, { charEncoding: "percent" });
    }).join("&");
    const path = (0, url_js_1.pathToFunc)("/api/generate/llm")();
    const headers = new Headers({
        "Content-Type": "application/x-www-form-urlencoded",
        Accept: "application/json",
    });
    const secConfig = await (0, security_js_1.extractSecurity)(client._options.apiKey);
    const securityInput = secConfig == null ? {} : { apiKey: secConfig };
    const requestSecurity = (0, security_js_1.resolveGlobalSecurity)(securityInput);
    const context = {
        operationID: "genLLM",
        oAuth2Scopes: [],
        resolvedSecurity: requestSecurity,
        securitySource: client._options.apiKey,
        retryConfig: (options === null || options === void 0 ? void 0 : options.retries)
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: (options === null || options === void 0 ? void 0 : options.retryCodes) || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "POST",
        path: path,
        headers: headers,
        body: body,
        timeoutMs: (options === null || options === void 0 ? void 0 : options.timeoutMs) || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return requestRes;
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: ["400", "401", "422", "4XX", "500", "5XX"],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return doResult;
    }
    const response = doResult.value;
    const responseFields = {
        ContentType: (_a = response.headers.get("content-type")) !== null && _a !== void 0 ? _a : "application/octet-stream",
        StatusCode: response.status,
        RawResponse: response,
        Headers: {},
    };
    const [result] = await M.match(M.json(200, operations.GenLLMResponse$inboundSchema, {
        key: "LLMResponse",
    }), M.jsonErr(400, errors.GenLLMResponseBody$inboundSchema), M.jsonErr(401, errors.GenLLMGenerateResponseBody$inboundSchema), M.jsonErr(422, errors.GenLLMGenerateResponseResponseBody$inboundSchema), M.jsonErr(500, errors.GenLLMGenerateResponse500ResponseBody$inboundSchema), M.fail(["4XX", "5XX"]), M.json("default", operations.GenLLMResponse$inboundSchema, {
        key: "studio-api-error",
    }))(response, { extraFields: responseFields });
    if (!result.ok) {
        return result;
    }
    return result;
}
