// livepeer/models/operations/addmultistreamtarget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddMultistreamTargetResponse$ = exports.AddMultistreamTargetResponse$outboundSchema = exports.AddMultistreamTargetResponse$inboundSchema = exports.AddMultistreamTargetRequest$ = exports.AddMultistreamTargetRequest$outboundSchema = exports.AddMultistreamTargetRequest$inboundSchema = void 0;
exports.addMultistreamTargetRequestToJSON = addMultistreamTargetRequestToJSON;
exports.addMultistreamTargetRequestFromJSON = addMultistreamTargetRequestFromJSON;
exports.addMultistreamTargetResponseToJSON = addMultistreamTargetResponseToJSON;
exports.addMultistreamTargetResponseFromJSON = addMultistreamTargetResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AddMultistreamTargetRequest$inboundSchema = z.object({
    id: z.string(),
    "target-add-payload": components.TargetAddPayload$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "target-add-payload": "targetAddPayload",
    });
});
/** @internal */
exports.AddMultistreamTargetRequest$outboundSchema = z.object({
    id: z.string(),
    targetAddPayload: components.TargetAddPayload$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        targetAddPayload: "target-add-payload",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddMultistreamTargetRequest$;
(function (AddMultistreamTargetRequest$) {
    /** @deprecated use `AddMultistreamTargetRequest$inboundSchema` instead. */
    AddMultistreamTargetRequest$.inboundSchema = exports.AddMultistreamTargetRequest$inboundSchema;
    /** @deprecated use `AddMultistreamTargetRequest$outboundSchema` instead. */
    AddMultistreamTargetRequest$.outboundSchema = exports.AddMultistreamTargetRequest$outboundSchema;
})(AddMultistreamTargetRequest$ || (exports.AddMultistreamTargetRequest$ = AddMultistreamTargetRequest$ = {}));
function addMultistreamTargetRequestToJSON(addMultistreamTargetRequest) {
    return JSON.stringify(exports.AddMultistreamTargetRequest$outboundSchema.parse(addMultistreamTargetRequest));
}
function addMultistreamTargetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddMultistreamTargetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddMultistreamTargetRequest' from JSON`);
}
/** @internal */
exports.AddMultistreamTargetResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.AddMultistreamTargetResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddMultistreamTargetResponse$;
(function (AddMultistreamTargetResponse$) {
    /** @deprecated use `AddMultistreamTargetResponse$inboundSchema` instead. */
    AddMultistreamTargetResponse$.inboundSchema = exports.AddMultistreamTargetResponse$inboundSchema;
    /** @deprecated use `AddMultistreamTargetResponse$outboundSchema` instead. */
    AddMultistreamTargetResponse$.outboundSchema = exports.AddMultistreamTargetResponse$outboundSchema;
})(AddMultistreamTargetResponse$ || (exports.AddMultistreamTargetResponse$ = AddMultistreamTargetResponse$ = {}));
function addMultistreamTargetResponseToJSON(addMultistreamTargetResponse) {
    return JSON.stringify(exports.AddMultistreamTargetResponse$outboundSchema.parse(addMultistreamTargetResponse));
}
function addMultistreamTargetResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddMultistreamTargetResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddMultistreamTargetResponse' from JSON`);
}
