// livepeer/models/operations/createsigningkey.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSigningKeyResponse$ = exports.CreateSigningKeyResponse$outboundSchema = exports.CreateSigningKeyResponse$inboundSchema = void 0;
exports.createSigningKeyResponseToJSON = createSigningKeyResponseToJSON;
exports.createSigningKeyResponseFromJSON = createSigningKeyResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.CreateSigningKeyResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    "signing-key": components.SigningKey$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
        "signing-key": "signingKey",
    });
});
/** @internal */
exports.CreateSigningKeyResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    signingKey: components.SigningKey$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
        signingKey: "signing-key",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateSigningKeyResponse$;
(function (CreateSigningKeyResponse$) {
    /** @deprecated use `CreateSigningKeyResponse$inboundSchema` instead. */
    CreateSigningKeyResponse$.inboundSchema = exports.CreateSigningKeyResponse$inboundSchema;
    /** @deprecated use `CreateSigningKeyResponse$outboundSchema` instead. */
    CreateSigningKeyResponse$.outboundSchema = exports.CreateSigningKeyResponse$outboundSchema;
})(CreateSigningKeyResponse$ || (exports.CreateSigningKeyResponse$ = CreateSigningKeyResponse$ = {}));
function createSigningKeyResponseToJSON(createSigningKeyResponse) {
    return JSON.stringify(exports.CreateSigningKeyResponse$outboundSchema.parse(createSigningKeyResponse));
}
function createSigningKeyResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateSigningKeyResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSigningKeyResponse' from JSON`);
}
