// livepeer/models/operations/createmultistreamtarget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateMultistreamTargetResponse$ = exports.CreateMultistreamTargetResponse$outboundSchema = exports.CreateMultistreamTargetResponse$inboundSchema = void 0;
exports.createMultistreamTargetResponseToJSON = createMultistreamTargetResponseToJSON;
exports.createMultistreamTargetResponseFromJSON = createMultistreamTargetResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.CreateMultistreamTargetResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    "multistream-target": components.MultistreamTarget$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
        "multistream-target": "multistreamTarget",
    });
});
/** @internal */
exports.CreateMultistreamTargetResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    multistreamTarget: components.MultistreamTarget$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
        multistreamTarget: "multistream-target",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateMultistreamTargetResponse$;
(function (CreateMultistreamTargetResponse$) {
    /** @deprecated use `CreateMultistreamTargetResponse$inboundSchema` instead. */
    CreateMultistreamTargetResponse$.inboundSchema = exports.CreateMultistreamTargetResponse$inboundSchema;
    /** @deprecated use `CreateMultistreamTargetResponse$outboundSchema` instead. */
    CreateMultistreamTargetResponse$.outboundSchema = exports.CreateMultistreamTargetResponse$outboundSchema;
})(CreateMultistreamTargetResponse$ || (exports.CreateMultistreamTargetResponse$ = CreateMultistreamTargetResponse$ = {}));
function createMultistreamTargetResponseToJSON(createMultistreamTargetResponse) {
    return JSON.stringify(exports.CreateMultistreamTargetResponse$outboundSchema.parse(createMultistreamTargetResponse));
}
function createMultistreamTargetResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateMultistreamTargetResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateMultistreamTargetResponse' from JSON`);
}
