// livepeer/models/operations/deletemultistreamtarget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteMultistreamTargetResponse$ = exports.DeleteMultistreamTargetResponse$outboundSchema = exports.DeleteMultistreamTargetResponse$inboundSchema = exports.DeleteMultistreamTargetRequest$ = exports.DeleteMultistreamTargetRequest$outboundSchema = exports.DeleteMultistreamTargetRequest$inboundSchema = void 0;
exports.deleteMultistreamTargetRequestToJSON = deleteMultistreamTargetRequestToJSON;
exports.deleteMultistreamTargetRequestFromJSON = deleteMultistreamTargetRequestFromJSON;
exports.deleteMultistreamTargetResponseToJSON = deleteMultistreamTargetResponseToJSON;
exports.deleteMultistreamTargetResponseFromJSON = deleteMultistreamTargetResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.DeleteMultistreamTargetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.DeleteMultistreamTargetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteMultistreamTargetRequest$;
(function (DeleteMultistreamTargetRequest$) {
    /** @deprecated use `DeleteMultistreamTargetRequest$inboundSchema` instead. */
    DeleteMultistreamTargetRequest$.inboundSchema = exports.DeleteMultistreamTargetRequest$inboundSchema;
    /** @deprecated use `DeleteMultistreamTargetRequest$outboundSchema` instead. */
    DeleteMultistreamTargetRequest$.outboundSchema = exports.DeleteMultistreamTargetRequest$outboundSchema;
})(DeleteMultistreamTargetRequest$ || (exports.DeleteMultistreamTargetRequest$ = DeleteMultistreamTargetRequest$ = {}));
function deleteMultistreamTargetRequestToJSON(deleteMultistreamTargetRequest) {
    return JSON.stringify(exports.DeleteMultistreamTargetRequest$outboundSchema.parse(deleteMultistreamTargetRequest));
}
function deleteMultistreamTargetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteMultistreamTargetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteMultistreamTargetRequest' from JSON`);
}
/** @internal */
exports.DeleteMultistreamTargetResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.DeleteMultistreamTargetResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteMultistreamTargetResponse$;
(function (DeleteMultistreamTargetResponse$) {
    /** @deprecated use `DeleteMultistreamTargetResponse$inboundSchema` instead. */
    DeleteMultistreamTargetResponse$.inboundSchema = exports.DeleteMultistreamTargetResponse$inboundSchema;
    /** @deprecated use `DeleteMultistreamTargetResponse$outboundSchema` instead. */
    DeleteMultistreamTargetResponse$.outboundSchema = exports.DeleteMultistreamTargetResponse$outboundSchema;
})(DeleteMultistreamTargetResponse$ || (exports.DeleteMultistreamTargetResponse$ = DeleteMultistreamTargetResponse$ = {}));
function deleteMultistreamTargetResponseToJSON(deleteMultistreamTargetResponse) {
    return JSON.stringify(exports.DeleteMultistreamTargetResponse$outboundSchema.parse(deleteMultistreamTargetResponse));
}
function deleteMultistreamTargetResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteMultistreamTargetResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteMultistreamTargetResponse' from JSON`);
}
