// livepeer/models/operations/deleteroomuser.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteRoomUserResponse$ = exports.DeleteRoomUserResponse$outboundSchema = exports.DeleteRoomUserResponse$inboundSchema = exports.DeleteRoomUserRequest$ = exports.DeleteRoomUserRequest$outboundSchema = exports.DeleteRoomUserRequest$inboundSchema = void 0;
exports.deleteRoomUserRequestToJSON = deleteRoomUserRequestToJSON;
exports.deleteRoomUserRequestFromJSON = deleteRoomUserRequestFromJSON;
exports.deleteRoomUserResponseToJSON = deleteRoomUserResponseToJSON;
exports.deleteRoomUserResponseFromJSON = deleteRoomUserResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.DeleteRoomUserRequest$inboundSchema = z.object({
    id: z.string(),
    userId: z.string(),
});
/** @internal */
exports.DeleteRoomUserRequest$outboundSchema = z.object({
    id: z.string(),
    userId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteRoomUserRequest$;
(function (DeleteRoomUserRequest$) {
    /** @deprecated use `DeleteRoomUserRequest$inboundSchema` instead. */
    DeleteRoomUserRequest$.inboundSchema = exports.DeleteRoomUserRequest$inboundSchema;
    /** @deprecated use `DeleteRoomUserRequest$outboundSchema` instead. */
    DeleteRoomUserRequest$.outboundSchema = exports.DeleteRoomUserRequest$outboundSchema;
})(DeleteRoomUserRequest$ || (exports.DeleteRoomUserRequest$ = DeleteRoomUserRequest$ = {}));
function deleteRoomUserRequestToJSON(deleteRoomUserRequest) {
    return JSON.stringify(exports.DeleteRoomUserRequest$outboundSchema.parse(deleteRoomUserRequest));
}
function deleteRoomUserRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteRoomUserRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteRoomUserRequest' from JSON`);
}
/** @internal */
exports.DeleteRoomUserResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.DeleteRoomUserResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteRoomUserResponse$;
(function (DeleteRoomUserResponse$) {
    /** @deprecated use `DeleteRoomUserResponse$inboundSchema` instead. */
    DeleteRoomUserResponse$.inboundSchema = exports.DeleteRoomUserResponse$inboundSchema;
    /** @deprecated use `DeleteRoomUserResponse$outboundSchema` instead. */
    DeleteRoomUserResponse$.outboundSchema = exports.DeleteRoomUserResponse$outboundSchema;
})(DeleteRoomUserResponse$ || (exports.DeleteRoomUserResponse$ = DeleteRoomUserResponse$ = {}));
function deleteRoomUserResponseToJSON(deleteRoomUserResponse) {
    return JSON.stringify(exports.DeleteRoomUserResponse$outboundSchema.parse(deleteRoomUserResponse));
}
function deleteRoomUserResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteRoomUserResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteRoomUserResponse' from JSON`);
}
