// livepeer/models/operations/deletesigningkey.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteSigningKeyResponse$ = exports.DeleteSigningKeyResponse$outboundSchema = exports.DeleteSigningKeyResponse$inboundSchema = exports.DeleteSigningKeyRequest$ = exports.DeleteSigningKeyRequest$outboundSchema = exports.DeleteSigningKeyRequest$inboundSchema = void 0;
exports.deleteSigningKeyRequestToJSON = deleteSigningKeyRequestToJSON;
exports.deleteSigningKeyRequestFromJSON = deleteSigningKeyRequestFromJSON;
exports.deleteSigningKeyResponseToJSON = deleteSigningKeyResponseToJSON;
exports.deleteSigningKeyResponseFromJSON = deleteSigningKeyResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.DeleteSigningKeyRequest$inboundSchema = z.object({
    keyId: z.string(),
});
/** @internal */
exports.DeleteSigningKeyRequest$outboundSchema = z.object({
    keyId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteSigningKeyRequest$;
(function (DeleteSigningKeyRequest$) {
    /** @deprecated use `DeleteSigningKeyRequest$inboundSchema` instead. */
    DeleteSigningKeyRequest$.inboundSchema = exports.DeleteSigningKeyRequest$inboundSchema;
    /** @deprecated use `DeleteSigningKeyRequest$outboundSchema` instead. */
    DeleteSigningKeyRequest$.outboundSchema = exports.DeleteSigningKeyRequest$outboundSchema;
})(DeleteSigningKeyRequest$ || (exports.DeleteSigningKeyRequest$ = DeleteSigningKeyRequest$ = {}));
function deleteSigningKeyRequestToJSON(deleteSigningKeyRequest) {
    return JSON.stringify(exports.DeleteSigningKeyRequest$outboundSchema.parse(deleteSigningKeyRequest));
}
function deleteSigningKeyRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteSigningKeyRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteSigningKeyRequest' from JSON`);
}
/** @internal */
exports.DeleteSigningKeyResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.DeleteSigningKeyResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteSigningKeyResponse$;
(function (DeleteSigningKeyResponse$) {
    /** @deprecated use `DeleteSigningKeyResponse$inboundSchema` instead. */
    DeleteSigningKeyResponse$.inboundSchema = exports.DeleteSigningKeyResponse$inboundSchema;
    /** @deprecated use `DeleteSigningKeyResponse$outboundSchema` instead. */
    DeleteSigningKeyResponse$.outboundSchema = exports.DeleteSigningKeyResponse$outboundSchema;
})(DeleteSigningKeyResponse$ || (exports.DeleteSigningKeyResponse$ = DeleteSigningKeyResponse$ = {}));
function deleteSigningKeyResponseToJSON(deleteSigningKeyResponse) {
    return JSON.stringify(exports.DeleteSigningKeyResponse$outboundSchema.parse(deleteSigningKeyResponse));
}
function deleteSigningKeyResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteSigningKeyResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteSigningKeyResponse' from JSON`);
}
