// livepeer/models/operations/deletewebhook.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteWebhookResponse$ = exports.DeleteWebhookResponse$outboundSchema = exports.DeleteWebhookResponse$inboundSchema = exports.DeleteWebhookRequest$ = exports.DeleteWebhookRequest$outboundSchema = exports.DeleteWebhookRequest$inboundSchema = void 0;
exports.deleteWebhookRequestToJSON = deleteWebhookRequestToJSON;
exports.deleteWebhookRequestFromJSON = deleteWebhookRequestFromJSON;
exports.deleteWebhookResponseToJSON = deleteWebhookResponseToJSON;
exports.deleteWebhookResponseFromJSON = deleteWebhookResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.DeleteWebhookRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.DeleteWebhookRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteWebhookRequest$;
(function (DeleteWebhookRequest$) {
    /** @deprecated use `DeleteWebhookRequest$inboundSchema` instead. */
    DeleteWebhookRequest$.inboundSchema = exports.DeleteWebhookRequest$inboundSchema;
    /** @deprecated use `DeleteWebhookRequest$outboundSchema` instead. */
    DeleteWebhookRequest$.outboundSchema = exports.DeleteWebhookRequest$outboundSchema;
})(DeleteWebhookRequest$ || (exports.DeleteWebhookRequest$ = DeleteWebhookRequest$ = {}));
function deleteWebhookRequestToJSON(deleteWebhookRequest) {
    return JSON.stringify(exports.DeleteWebhookRequest$outboundSchema.parse(deleteWebhookRequest));
}
function deleteWebhookRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteWebhookRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteWebhookRequest' from JSON`);
}
/** @internal */
exports.DeleteWebhookResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    webhook: components.Webhook$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.DeleteWebhookResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    webhook: components.Webhook$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteWebhookResponse$;
(function (DeleteWebhookResponse$) {
    /** @deprecated use `DeleteWebhookResponse$inboundSchema` instead. */
    DeleteWebhookResponse$.inboundSchema = exports.DeleteWebhookResponse$inboundSchema;
    /** @deprecated use `DeleteWebhookResponse$outboundSchema` instead. */
    DeleteWebhookResponse$.outboundSchema = exports.DeleteWebhookResponse$outboundSchema;
})(DeleteWebhookResponse$ || (exports.DeleteWebhookResponse$ = DeleteWebhookResponse$ = {}));
function deleteWebhookResponseToJSON(deleteWebhookResponse) {
    return JSON.stringify(exports.DeleteWebhookResponse$outboundSchema.parse(deleteWebhookResponse));
}
function deleteWebhookResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteWebhookResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteWebhookResponse' from JSON`);
}
