// livepeer/models/operations/deletestream.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteStreamResponse$ = exports.DeleteStreamResponse$outboundSchema = exports.DeleteStreamResponse$inboundSchema = exports.DeleteStreamRequest$ = exports.DeleteStreamRequest$outboundSchema = exports.DeleteStreamRequest$inboundSchema = void 0;
exports.deleteStreamRequestToJSON = deleteStreamRequestToJSON;
exports.deleteStreamRequestFromJSON = deleteStreamRequestFromJSON;
exports.deleteStreamResponseToJSON = deleteStreamResponseToJSON;
exports.deleteStreamResponseFromJSON = deleteStreamResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.DeleteStreamRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.DeleteStreamRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteStreamRequest$;
(function (DeleteStreamRequest$) {
    /** @deprecated use `DeleteStreamRequest$inboundSchema` instead. */
    DeleteStreamRequest$.inboundSchema = exports.DeleteStreamRequest$inboundSchema;
    /** @deprecated use `DeleteStreamRequest$outboundSchema` instead. */
    DeleteStreamRequest$.outboundSchema = exports.DeleteStreamRequest$outboundSchema;
})(DeleteStreamRequest$ || (exports.DeleteStreamRequest$ = DeleteStreamRequest$ = {}));
function deleteStreamRequestToJSON(deleteStreamRequest) {
    return JSON.stringify(exports.DeleteStreamRequest$outboundSchema.parse(deleteStreamRequest));
}
function deleteStreamRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteStreamRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteStreamRequest' from JSON`);
}
/** @internal */
exports.DeleteStreamResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.DeleteStreamResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteStreamResponse$;
(function (DeleteStreamResponse$) {
    /** @deprecated use `DeleteStreamResponse$inboundSchema` instead. */
    DeleteStreamResponse$.inboundSchema = exports.DeleteStreamResponse$inboundSchema;
    /** @deprecated use `DeleteStreamResponse$outboundSchema` instead. */
    DeleteStreamResponse$.outboundSchema = exports.DeleteStreamResponse$outboundSchema;
})(DeleteStreamResponse$ || (exports.DeleteStreamResponse$ = DeleteStreamResponse$ = {}));
function deleteStreamResponseToJSON(deleteStreamResponse) {
    return JSON.stringify(exports.DeleteStreamResponse$outboundSchema.parse(deleteStreamResponse));
}
function deleteStreamResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteStreamResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteStreamResponse' from JSON`);
}
