// livepeer/models/operations/deleteroom.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteRoomResponse$ = exports.DeleteRoomResponse$outboundSchema = exports.DeleteRoomResponse$inboundSchema = exports.DeleteRoomRequest$ = exports.DeleteRoomRequest$outboundSchema = exports.DeleteRoomRequest$inboundSchema = void 0;
exports.deleteRoomRequestToJSON = deleteRoomRequestToJSON;
exports.deleteRoomRequestFromJSON = deleteRoomRequestFromJSON;
exports.deleteRoomResponseToJSON = deleteRoomResponseToJSON;
exports.deleteRoomResponseFromJSON = deleteRoomResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.DeleteRoomRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.DeleteRoomRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteRoomRequest$;
(function (DeleteRoomRequest$) {
    /** @deprecated use `DeleteRoomRequest$inboundSchema` instead. */
    DeleteRoomRequest$.inboundSchema = exports.DeleteRoomRequest$inboundSchema;
    /** @deprecated use `DeleteRoomRequest$outboundSchema` instead. */
    DeleteRoomRequest$.outboundSchema = exports.DeleteRoomRequest$outboundSchema;
})(DeleteRoomRequest$ || (exports.DeleteRoomRequest$ = DeleteRoomRequest$ = {}));
function deleteRoomRequestToJSON(deleteRoomRequest) {
    return JSON.stringify(exports.DeleteRoomRequest$outboundSchema.parse(deleteRoomRequest));
}
function deleteRoomRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteRoomRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteRoomRequest' from JSON`);
}
/** @internal */
exports.DeleteRoomResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.DeleteRoomResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteRoomResponse$;
(function (DeleteRoomResponse$) {
    /** @deprecated use `DeleteRoomResponse$inboundSchema` instead. */
    DeleteRoomResponse$.inboundSchema = exports.DeleteRoomResponse$inboundSchema;
    /** @deprecated use `DeleteRoomResponse$outboundSchema` instead. */
    DeleteRoomResponse$.outboundSchema = exports.DeleteRoomResponse$outboundSchema;
})(DeleteRoomResponse$ || (exports.DeleteRoomResponse$ = DeleteRoomResponse$ = {}));
function deleteRoomResponseToJSON(deleteRoomResponse) {
    return JSON.stringify(exports.DeleteRoomResponse$outboundSchema.parse(deleteRoomResponse));
}
function deleteRoomResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteRoomResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteRoomResponse' from JSON`);
}
