// livepeer/models/operations/deleteasset.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteAssetResponse$ = exports.DeleteAssetResponse$outboundSchema = exports.DeleteAssetResponse$inboundSchema = exports.DeleteAssetRequest$ = exports.DeleteAssetRequest$outboundSchema = exports.DeleteAssetRequest$inboundSchema = void 0;
exports.deleteAssetRequestToJSON = deleteAssetRequestToJSON;
exports.deleteAssetRequestFromJSON = deleteAssetRequestFromJSON;
exports.deleteAssetResponseToJSON = deleteAssetResponseToJSON;
exports.deleteAssetResponseFromJSON = deleteAssetResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.DeleteAssetRequest$inboundSchema = z.object({
    assetId: z.string(),
});
/** @internal */
exports.DeleteAssetRequest$outboundSchema = z.object({
    assetId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteAssetRequest$;
(function (DeleteAssetRequest$) {
    /** @deprecated use `DeleteAssetRequest$inboundSchema` instead. */
    DeleteAssetRequest$.inboundSchema = exports.DeleteAssetRequest$inboundSchema;
    /** @deprecated use `DeleteAssetRequest$outboundSchema` instead. */
    DeleteAssetRequest$.outboundSchema = exports.DeleteAssetRequest$outboundSchema;
})(DeleteAssetRequest$ || (exports.DeleteAssetRequest$ = DeleteAssetRequest$ = {}));
function deleteAssetRequestToJSON(deleteAssetRequest) {
    return JSON.stringify(exports.DeleteAssetRequest$outboundSchema.parse(deleteAssetRequest));
}
function deleteAssetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteAssetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteAssetRequest' from JSON`);
}
/** @internal */
exports.DeleteAssetResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.DeleteAssetResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteAssetResponse$;
(function (DeleteAssetResponse$) {
    /** @deprecated use `DeleteAssetResponse$inboundSchema` instead. */
    DeleteAssetResponse$.inboundSchema = exports.DeleteAssetResponse$inboundSchema;
    /** @deprecated use `DeleteAssetResponse$outboundSchema` instead. */
    DeleteAssetResponse$.outboundSchema = exports.DeleteAssetResponse$outboundSchema;
})(DeleteAssetResponse$ || (exports.DeleteAssetResponse$ = DeleteAssetResponse$ = {}));
function deleteAssetResponseToJSON(deleteAssetResponse) {
    return JSON.stringify(exports.DeleteAssetResponse$outboundSchema.parse(deleteAssetResponse));
}
function deleteAssetResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteAssetResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteAssetResponse' from JSON`);
}
