// livepeer/models/operations/gensegmentanything2.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenSegmentAnything2Response$ = exports.GenSegmentAnything2Response$outboundSchema = exports.GenSegmentAnything2Response$inboundSchema = void 0;
exports.genSegmentAnything2ResponseToJSON = genSegmentAnything2ResponseToJSON;
exports.genSegmentAnything2ResponseFromJSON = genSegmentAnything2ResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
const errors = __importStar(require("../errors/index.js"));
/** @internal */
exports.GenSegmentAnything2Response$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    MasksResponse: components.MasksResponse$inboundSchema.optional(),
    "studio-api-error": errors.StudioApiError$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
        "MasksResponse": "masksResponse",
        "studio-api-error": "studioApiError",
    });
});
/** @internal */
exports.GenSegmentAnything2Response$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    masksResponse: components.MasksResponse$outboundSchema.optional(),
    studioApiError: errors.StudioApiError$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
        masksResponse: "MasksResponse",
        studioApiError: "studio-api-error",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenSegmentAnything2Response$;
(function (GenSegmentAnything2Response$) {
    /** @deprecated use `GenSegmentAnything2Response$inboundSchema` instead. */
    GenSegmentAnything2Response$.inboundSchema = exports.GenSegmentAnything2Response$inboundSchema;
    /** @deprecated use `GenSegmentAnything2Response$outboundSchema` instead. */
    GenSegmentAnything2Response$.outboundSchema = exports.GenSegmentAnything2Response$outboundSchema;
})(GenSegmentAnything2Response$ || (exports.GenSegmentAnything2Response$ = GenSegmentAnything2Response$ = {}));
function genSegmentAnything2ResponseToJSON(genSegmentAnything2Response) {
    return JSON.stringify(exports.GenSegmentAnything2Response$outboundSchema.parse(genSegmentAnything2Response));
}
function genSegmentAnything2ResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenSegmentAnything2Response$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenSegmentAnything2Response' from JSON`);
}
