// livepeer/models/operations/genupscale.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenUpscaleResponse$ = exports.GenUpscaleResponse$outboundSchema = exports.GenUpscaleResponse$inboundSchema = void 0;
exports.genUpscaleResponseToJSON = genUpscaleResponseToJSON;
exports.genUpscaleResponseFromJSON = genUpscaleResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
const errors = __importStar(require("../errors/index.js"));
/** @internal */
exports.GenUpscaleResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    ImageResponse: components.ImageResponse$inboundSchema.optional(),
    "studio-api-error": errors.StudioApiError$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
        "ImageResponse": "imageResponse",
        "studio-api-error": "studioApiError",
    });
});
/** @internal */
exports.GenUpscaleResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    imageResponse: components.ImageResponse$outboundSchema.optional(),
    studioApiError: errors.StudioApiError$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
        imageResponse: "ImageResponse",
        studioApiError: "studio-api-error",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenUpscaleResponse$;
(function (GenUpscaleResponse$) {
    /** @deprecated use `GenUpscaleResponse$inboundSchema` instead. */
    GenUpscaleResponse$.inboundSchema = exports.GenUpscaleResponse$inboundSchema;
    /** @deprecated use `GenUpscaleResponse$outboundSchema` instead. */
    GenUpscaleResponse$.outboundSchema = exports.GenUpscaleResponse$outboundSchema;
})(GenUpscaleResponse$ || (exports.GenUpscaleResponse$ = GenUpscaleResponse$ = {}));
function genUpscaleResponseToJSON(genUpscaleResponse) {
    return JSON.stringify(exports.GenUpscaleResponse$outboundSchema.parse(genUpscaleResponse));
}
function genUpscaleResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenUpscaleResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenUpscaleResponse' from JSON`);
}
