// livepeer/models/operations/getasset.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAssetResponse$ = exports.GetAssetResponse$outboundSchema = exports.GetAssetResponse$inboundSchema = exports.GetAssetRequest$ = exports.GetAssetRequest$outboundSchema = exports.GetAssetRequest$inboundSchema = void 0;
exports.getAssetRequestToJSON = getAssetRequestToJSON;
exports.getAssetRequestFromJSON = getAssetRequestFromJSON;
exports.getAssetResponseToJSON = getAssetResponseToJSON;
exports.getAssetResponseFromJSON = getAssetResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetAssetRequest$inboundSchema = z.object({
    assetId: z.string(),
});
/** @internal */
exports.GetAssetRequest$outboundSchema = z.object({
    assetId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAssetRequest$;
(function (GetAssetRequest$) {
    /** @deprecated use `GetAssetRequest$inboundSchema` instead. */
    GetAssetRequest$.inboundSchema = exports.GetAssetRequest$inboundSchema;
    /** @deprecated use `GetAssetRequest$outboundSchema` instead. */
    GetAssetRequest$.outboundSchema = exports.GetAssetRequest$outboundSchema;
})(GetAssetRequest$ || (exports.GetAssetRequest$ = GetAssetRequest$ = {}));
function getAssetRequestToJSON(getAssetRequest) {
    return JSON.stringify(exports.GetAssetRequest$outboundSchema.parse(getAssetRequest));
}
function getAssetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetAssetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAssetRequest' from JSON`);
}
/** @internal */
exports.GetAssetResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    asset: components.Asset$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetAssetResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    asset: components.Asset$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAssetResponse$;
(function (GetAssetResponse$) {
    /** @deprecated use `GetAssetResponse$inboundSchema` instead. */
    GetAssetResponse$.inboundSchema = exports.GetAssetResponse$inboundSchema;
    /** @deprecated use `GetAssetResponse$outboundSchema` instead. */
    GetAssetResponse$.outboundSchema = exports.GetAssetResponse$outboundSchema;
})(GetAssetResponse$ || (exports.GetAssetResponse$ = GetAssetResponse$ = {}));
function getAssetResponseToJSON(getAssetResponse) {
    return JSON.stringify(exports.GetAssetResponse$outboundSchema.parse(getAssetResponse));
}
function getAssetResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetAssetResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAssetResponse' from JSON`);
}
