// livepeer/models/operations/getplaybackinfo.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetPlaybackInfoResponse$ = exports.GetPlaybackInfoResponse$outboundSchema = exports.GetPlaybackInfoResponse$inboundSchema = exports.GetPlaybackInfoRequest$ = exports.GetPlaybackInfoRequest$outboundSchema = exports.GetPlaybackInfoRequest$inboundSchema = void 0;
exports.getPlaybackInfoRequestToJSON = getPlaybackInfoRequestToJSON;
exports.getPlaybackInfoRequestFromJSON = getPlaybackInfoRequestFromJSON;
exports.getPlaybackInfoResponseToJSON = getPlaybackInfoResponseToJSON;
exports.getPlaybackInfoResponseFromJSON = getPlaybackInfoResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetPlaybackInfoRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.GetPlaybackInfoRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetPlaybackInfoRequest$;
(function (GetPlaybackInfoRequest$) {
    /** @deprecated use `GetPlaybackInfoRequest$inboundSchema` instead. */
    GetPlaybackInfoRequest$.inboundSchema = exports.GetPlaybackInfoRequest$inboundSchema;
    /** @deprecated use `GetPlaybackInfoRequest$outboundSchema` instead. */
    GetPlaybackInfoRequest$.outboundSchema = exports.GetPlaybackInfoRequest$outboundSchema;
})(GetPlaybackInfoRequest$ || (exports.GetPlaybackInfoRequest$ = GetPlaybackInfoRequest$ = {}));
function getPlaybackInfoRequestToJSON(getPlaybackInfoRequest) {
    return JSON.stringify(exports.GetPlaybackInfoRequest$outboundSchema.parse(getPlaybackInfoRequest));
}
function getPlaybackInfoRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetPlaybackInfoRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetPlaybackInfoRequest' from JSON`);
}
/** @internal */
exports.GetPlaybackInfoResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    "playback-info": components.PlaybackInfo$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
        "playback-info": "playbackInfo",
    });
});
/** @internal */
exports.GetPlaybackInfoResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    playbackInfo: components.PlaybackInfo$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
        playbackInfo: "playback-info",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetPlaybackInfoResponse$;
(function (GetPlaybackInfoResponse$) {
    /** @deprecated use `GetPlaybackInfoResponse$inboundSchema` instead. */
    GetPlaybackInfoResponse$.inboundSchema = exports.GetPlaybackInfoResponse$inboundSchema;
    /** @deprecated use `GetPlaybackInfoResponse$outboundSchema` instead. */
    GetPlaybackInfoResponse$.outboundSchema = exports.GetPlaybackInfoResponse$outboundSchema;
})(GetPlaybackInfoResponse$ || (exports.GetPlaybackInfoResponse$ = GetPlaybackInfoResponse$ = {}));
function getPlaybackInfoResponseToJSON(getPlaybackInfoResponse) {
    return JSON.stringify(exports.GetPlaybackInfoResponse$outboundSchema.parse(getPlaybackInfoResponse));
}
function getPlaybackInfoResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetPlaybackInfoResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetPlaybackInfoResponse' from JSON`);
}
