// livepeer/models/operations/getroom.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRoomResponse$ = exports.GetRoomResponse$outboundSchema = exports.GetRoomResponse$inboundSchema = exports.GetRoomRequest$ = exports.GetRoomRequest$outboundSchema = exports.GetRoomRequest$inboundSchema = void 0;
exports.getRoomRequestToJSON = getRoomRequestToJSON;
exports.getRoomRequestFromJSON = getRoomRequestFromJSON;
exports.getRoomResponseToJSON = getRoomResponseToJSON;
exports.getRoomResponseFromJSON = getRoomResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetRoomRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.GetRoomRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRoomRequest$;
(function (GetRoomRequest$) {
    /** @deprecated use `GetRoomRequest$inboundSchema` instead. */
    GetRoomRequest$.inboundSchema = exports.GetRoomRequest$inboundSchema;
    /** @deprecated use `GetRoomRequest$outboundSchema` instead. */
    GetRoomRequest$.outboundSchema = exports.GetRoomRequest$outboundSchema;
})(GetRoomRequest$ || (exports.GetRoomRequest$ = GetRoomRequest$ = {}));
function getRoomRequestToJSON(getRoomRequest) {
    return JSON.stringify(exports.GetRoomRequest$outboundSchema.parse(getRoomRequest));
}
function getRoomRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRoomRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRoomRequest' from JSON`);
}
/** @internal */
exports.GetRoomResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    room: components.Room$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetRoomResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    room: components.Room$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRoomResponse$;
(function (GetRoomResponse$) {
    /** @deprecated use `GetRoomResponse$inboundSchema` instead. */
    GetRoomResponse$.inboundSchema = exports.GetRoomResponse$inboundSchema;
    /** @deprecated use `GetRoomResponse$outboundSchema` instead. */
    GetRoomResponse$.outboundSchema = exports.GetRoomResponse$outboundSchema;
})(GetRoomResponse$ || (exports.GetRoomResponse$ = GetRoomResponse$ = {}));
function getRoomResponseToJSON(getRoomResponse) {
    return JSON.stringify(exports.GetRoomResponse$outboundSchema.parse(getRoomResponse));
}
function getRoomResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRoomResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRoomResponse' from JSON`);
}
