// livepeer/models/operations/getroomuser.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRoomUserResponse$ = exports.GetRoomUserResponse$outboundSchema = exports.GetRoomUserResponse$inboundSchema = exports.GetRoomUserRequest$ = exports.GetRoomUserRequest$outboundSchema = exports.GetRoomUserRequest$inboundSchema = void 0;
exports.getRoomUserRequestToJSON = getRoomUserRequestToJSON;
exports.getRoomUserRequestFromJSON = getRoomUserRequestFromJSON;
exports.getRoomUserResponseToJSON = getRoomUserResponseToJSON;
exports.getRoomUserResponseFromJSON = getRoomUserResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetRoomUserRequest$inboundSchema = z.object({
    id: z.string(),
    userId: z.string(),
});
/** @internal */
exports.GetRoomUserRequest$outboundSchema = z.object({
    id: z.string(),
    userId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRoomUserRequest$;
(function (GetRoomUserRequest$) {
    /** @deprecated use `GetRoomUserRequest$inboundSchema` instead. */
    GetRoomUserRequest$.inboundSchema = exports.GetRoomUserRequest$inboundSchema;
    /** @deprecated use `GetRoomUserRequest$outboundSchema` instead. */
    GetRoomUserRequest$.outboundSchema = exports.GetRoomUserRequest$outboundSchema;
})(GetRoomUserRequest$ || (exports.GetRoomUserRequest$ = GetRoomUserRequest$ = {}));
function getRoomUserRequestToJSON(getRoomUserRequest) {
    return JSON.stringify(exports.GetRoomUserRequest$outboundSchema.parse(getRoomUserRequest));
}
function getRoomUserRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRoomUserRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRoomUserRequest' from JSON`);
}
/** @internal */
exports.GetRoomUserResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    "get-room-user-response": components.GetRoomUserResponse$inboundSchema
        .optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
        "get-room-user-response": "getRoomUserResponse",
    });
});
/** @internal */
exports.GetRoomUserResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    getRoomUserResponse: components.GetRoomUserResponse$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
        getRoomUserResponse: "get-room-user-response",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRoomUserResponse$;
(function (GetRoomUserResponse$) {
    /** @deprecated use `GetRoomUserResponse$inboundSchema` instead. */
    GetRoomUserResponse$.inboundSchema = exports.GetRoomUserResponse$inboundSchema;
    /** @deprecated use `GetRoomUserResponse$outboundSchema` instead. */
    GetRoomUserResponse$.outboundSchema = exports.GetRoomUserResponse$outboundSchema;
})(GetRoomUserResponse$ || (exports.GetRoomUserResponse$ = GetRoomUserResponse$ = {}));
function getRoomUserResponseToJSON(getRoomUserResponse) {
    return JSON.stringify(exports.GetRoomUserResponse$outboundSchema.parse(getRoomUserResponse));
}
function getRoomUserResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRoomUserResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRoomUserResponse' from JSON`);
}
