// livepeer/models/operations/getrecordedsessions.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRecordedSessionsResponse$ = exports.GetRecordedSessionsResponse$outboundSchema = exports.GetRecordedSessionsResponse$inboundSchema = exports.GetRecordedSessionsRequest$ = exports.GetRecordedSessionsRequest$outboundSchema = exports.GetRecordedSessionsRequest$inboundSchema = exports.RecordT$ = exports.RecordT$outboundSchema = exports.RecordT$inboundSchema = void 0;
exports.recordTToJSON = recordTToJSON;
exports.recordTFromJSON = recordTFromJSON;
exports.getRecordedSessionsRequestToJSON = getRecordedSessionsRequestToJSON;
exports.getRecordedSessionsRequestFromJSON = getRecordedSessionsRequestFromJSON;
exports.getRecordedSessionsResponseToJSON = getRecordedSessionsResponseToJSON;
exports.getRecordedSessionsResponseFromJSON = getRecordedSessionsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.RecordT$inboundSchema = z.union([z.boolean(), z.number().int()]);
/** @internal */
exports.RecordT$outboundSchema = z.union([z.boolean(), z.number().int()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RecordT$;
(function (RecordT$) {
    /** @deprecated use `RecordT$inboundSchema` instead. */
    RecordT$.inboundSchema = exports.RecordT$inboundSchema;
    /** @deprecated use `RecordT$outboundSchema` instead. */
    RecordT$.outboundSchema = exports.RecordT$outboundSchema;
})(RecordT$ || (exports.RecordT$ = RecordT$ = {}));
function recordTToJSON(recordT) {
    return JSON.stringify(exports.RecordT$outboundSchema.parse(recordT));
}
function recordTFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RecordT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RecordT' from JSON`);
}
/** @internal */
exports.GetRecordedSessionsRequest$inboundSchema = z.object({
    parentId: z.string(),
    record: z.union([z.boolean(), z.number().int()]).optional(),
});
/** @internal */
exports.GetRecordedSessionsRequest$outboundSchema = z.object({
    parentId: z.string(),
    record: z.union([z.boolean(), z.number().int()]).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRecordedSessionsRequest$;
(function (GetRecordedSessionsRequest$) {
    /** @deprecated use `GetRecordedSessionsRequest$inboundSchema` instead. */
    GetRecordedSessionsRequest$.inboundSchema = exports.GetRecordedSessionsRequest$inboundSchema;
    /** @deprecated use `GetRecordedSessionsRequest$outboundSchema` instead. */
    GetRecordedSessionsRequest$.outboundSchema = exports.GetRecordedSessionsRequest$outboundSchema;
})(GetRecordedSessionsRequest$ || (exports.GetRecordedSessionsRequest$ = GetRecordedSessionsRequest$ = {}));
function getRecordedSessionsRequestToJSON(getRecordedSessionsRequest) {
    return JSON.stringify(exports.GetRecordedSessionsRequest$outboundSchema.parse(getRecordedSessionsRequest));
}
function getRecordedSessionsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRecordedSessionsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordedSessionsRequest' from JSON`);
}
/** @internal */
exports.GetRecordedSessionsResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    data: z.array(components.Session$inboundSchema).optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetRecordedSessionsResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    data: z.array(components.Session$outboundSchema).optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRecordedSessionsResponse$;
(function (GetRecordedSessionsResponse$) {
    /** @deprecated use `GetRecordedSessionsResponse$inboundSchema` instead. */
    GetRecordedSessionsResponse$.inboundSchema = exports.GetRecordedSessionsResponse$inboundSchema;
    /** @deprecated use `GetRecordedSessionsResponse$outboundSchema` instead. */
    GetRecordedSessionsResponse$.outboundSchema = exports.GetRecordedSessionsResponse$outboundSchema;
})(GetRecordedSessionsResponse$ || (exports.GetRecordedSessionsResponse$ = GetRecordedSessionsResponse$ = {}));
function getRecordedSessionsResponseToJSON(getRecordedSessionsResponse) {
    return JSON.stringify(exports.GetRecordedSessionsResponse$outboundSchema.parse(getRecordedSessionsResponse));
}
function getRecordedSessionsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRecordedSessionsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordedSessionsResponse' from JSON`);
}
