// livepeer/models/operations/getstream.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetStreamResponse$ = exports.GetStreamResponse$outboundSchema = exports.GetStreamResponse$inboundSchema = exports.GetStreamRequest$ = exports.GetStreamRequest$outboundSchema = exports.GetStreamRequest$inboundSchema = void 0;
exports.getStreamRequestToJSON = getStreamRequestToJSON;
exports.getStreamRequestFromJSON = getStreamRequestFromJSON;
exports.getStreamResponseToJSON = getStreamResponseToJSON;
exports.getStreamResponseFromJSON = getStreamResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetStreamRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.GetStreamRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetStreamRequest$;
(function (GetStreamRequest$) {
    /** @deprecated use `GetStreamRequest$inboundSchema` instead. */
    GetStreamRequest$.inboundSchema = exports.GetStreamRequest$inboundSchema;
    /** @deprecated use `GetStreamRequest$outboundSchema` instead. */
    GetStreamRequest$.outboundSchema = exports.GetStreamRequest$outboundSchema;
})(GetStreamRequest$ || (exports.GetStreamRequest$ = GetStreamRequest$ = {}));
function getStreamRequestToJSON(getStreamRequest) {
    return JSON.stringify(exports.GetStreamRequest$outboundSchema.parse(getStreamRequest));
}
function getStreamRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetStreamRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetStreamRequest' from JSON`);
}
/** @internal */
exports.GetStreamResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    stream: components.Stream$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetStreamResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    stream: components.Stream$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetStreamResponse$;
(function (GetStreamResponse$) {
    /** @deprecated use `GetStreamResponse$inboundSchema` instead. */
    GetStreamResponse$.inboundSchema = exports.GetStreamResponse$inboundSchema;
    /** @deprecated use `GetStreamResponse$outboundSchema` instead. */
    GetStreamResponse$.outboundSchema = exports.GetStreamResponse$outboundSchema;
})(GetStreamResponse$ || (exports.GetStreamResponse$ = GetStreamResponse$ = {}));
function getStreamResponseToJSON(getStreamResponse) {
    return JSON.stringify(exports.GetStreamResponse$outboundSchema.parse(getStreamResponse));
}
function getStreamResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetStreamResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetStreamResponse' from JSON`);
}
