// livepeer/models/operations/getstreams.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetStreamsResponse$ = exports.GetStreamsResponse$outboundSchema = exports.GetStreamsResponse$inboundSchema = exports.GetStreamsRequest$ = exports.GetStreamsRequest$outboundSchema = exports.GetStreamsRequest$inboundSchema = void 0;
exports.getStreamsRequestToJSON = getStreamsRequestToJSON;
exports.getStreamsRequestFromJSON = getStreamsRequestFromJSON;
exports.getStreamsResponseToJSON = getStreamsResponseToJSON;
exports.getStreamsResponseFromJSON = getStreamsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetStreamsRequest$inboundSchema = z.object({
    streamsonly: z.string().optional(),
});
/** @internal */
exports.GetStreamsRequest$outboundSchema = z.object({
    streamsonly: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetStreamsRequest$;
(function (GetStreamsRequest$) {
    /** @deprecated use `GetStreamsRequest$inboundSchema` instead. */
    GetStreamsRequest$.inboundSchema = exports.GetStreamsRequest$inboundSchema;
    /** @deprecated use `GetStreamsRequest$outboundSchema` instead. */
    GetStreamsRequest$.outboundSchema = exports.GetStreamsRequest$outboundSchema;
})(GetStreamsRequest$ || (exports.GetStreamsRequest$ = GetStreamsRequest$ = {}));
function getStreamsRequestToJSON(getStreamsRequest) {
    return JSON.stringify(exports.GetStreamsRequest$outboundSchema.parse(getStreamsRequest));
}
function getStreamsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetStreamsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetStreamsRequest' from JSON`);
}
/** @internal */
exports.GetStreamsResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    data: z.array(components.Stream$inboundSchema).optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetStreamsResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    data: z.array(components.Stream$outboundSchema).optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetStreamsResponse$;
(function (GetStreamsResponse$) {
    /** @deprecated use `GetStreamsResponse$inboundSchema` instead. */
    GetStreamsResponse$.inboundSchema = exports.GetStreamsResponse$inboundSchema;
    /** @deprecated use `GetStreamsResponse$outboundSchema` instead. */
    GetStreamsResponse$.outboundSchema = exports.GetStreamsResponse$outboundSchema;
})(GetStreamsResponse$ || (exports.GetStreamsResponse$ = GetStreamsResponse$ = {}));
function getStreamsResponseToJSON(getStreamsResponse) {
    return JSON.stringify(exports.GetStreamsResponse$outboundSchema.parse(getStreamsResponse));
}
function getStreamsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetStreamsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetStreamsResponse' from JSON`);
}
