// livepeer/models/operations/getsession.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSessionResponse$ = exports.GetSessionResponse$outboundSchema = exports.GetSessionResponse$inboundSchema = exports.GetSessionRequest$ = exports.GetSessionRequest$outboundSchema = exports.GetSessionRequest$inboundSchema = void 0;
exports.getSessionRequestToJSON = getSessionRequestToJSON;
exports.getSessionRequestFromJSON = getSessionRequestFromJSON;
exports.getSessionResponseToJSON = getSessionResponseToJSON;
exports.getSessionResponseFromJSON = getSessionResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetSessionRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.GetSessionRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetSessionRequest$;
(function (GetSessionRequest$) {
    /** @deprecated use `GetSessionRequest$inboundSchema` instead. */
    GetSessionRequest$.inboundSchema = exports.GetSessionRequest$inboundSchema;
    /** @deprecated use `GetSessionRequest$outboundSchema` instead. */
    GetSessionRequest$.outboundSchema = exports.GetSessionRequest$outboundSchema;
})(GetSessionRequest$ || (exports.GetSessionRequest$ = GetSessionRequest$ = {}));
function getSessionRequestToJSON(getSessionRequest) {
    return JSON.stringify(exports.GetSessionRequest$outboundSchema.parse(getSessionRequest));
}
function getSessionRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetSessionRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetSessionRequest' from JSON`);
}
/** @internal */
exports.GetSessionResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    session: components.Session$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetSessionResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    session: components.Session$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetSessionResponse$;
(function (GetSessionResponse$) {
    /** @deprecated use `GetSessionResponse$inboundSchema` instead. */
    GetSessionResponse$.inboundSchema = exports.GetSessionResponse$inboundSchema;
    /** @deprecated use `GetSessionResponse$outboundSchema` instead. */
    GetSessionResponse$.outboundSchema = exports.GetSessionResponse$outboundSchema;
})(GetSessionResponse$ || (exports.GetSessionResponse$ = GetSessionResponse$ = {}));
function getSessionResponseToJSON(getSessionResponse) {
    return JSON.stringify(exports.GetSessionResponse$outboundSchema.parse(getSessionResponse));
}
function getSessionResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetSessionResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetSessionResponse' from JSON`);
}
