// livepeer/models/operations/getsessionclips.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSessionClipsResponse$ = exports.GetSessionClipsResponse$outboundSchema = exports.GetSessionClipsResponse$inboundSchema = exports.GetSessionClipsRequest$ = exports.GetSessionClipsRequest$outboundSchema = exports.GetSessionClipsRequest$inboundSchema = void 0;
exports.getSessionClipsRequestToJSON = getSessionClipsRequestToJSON;
exports.getSessionClipsRequestFromJSON = getSessionClipsRequestFromJSON;
exports.getSessionClipsResponseToJSON = getSessionClipsResponseToJSON;
exports.getSessionClipsResponseFromJSON = getSessionClipsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetSessionClipsRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.GetSessionClipsRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetSessionClipsRequest$;
(function (GetSessionClipsRequest$) {
    /** @deprecated use `GetSessionClipsRequest$inboundSchema` instead. */
    GetSessionClipsRequest$.inboundSchema = exports.GetSessionClipsRequest$inboundSchema;
    /** @deprecated use `GetSessionClipsRequest$outboundSchema` instead. */
    GetSessionClipsRequest$.outboundSchema = exports.GetSessionClipsRequest$outboundSchema;
})(GetSessionClipsRequest$ || (exports.GetSessionClipsRequest$ = GetSessionClipsRequest$ = {}));
function getSessionClipsRequestToJSON(getSessionClipsRequest) {
    return JSON.stringify(exports.GetSessionClipsRequest$outboundSchema.parse(getSessionClipsRequest));
}
function getSessionClipsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetSessionClipsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetSessionClipsRequest' from JSON`);
}
/** @internal */
exports.GetSessionClipsResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    data: z.array(components.Asset$inboundSchema).optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetSessionClipsResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    data: z.array(components.Asset$outboundSchema).optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetSessionClipsResponse$;
(function (GetSessionClipsResponse$) {
    /** @deprecated use `GetSessionClipsResponse$inboundSchema` instead. */
    GetSessionClipsResponse$.inboundSchema = exports.GetSessionClipsResponse$inboundSchema;
    /** @deprecated use `GetSessionClipsResponse$outboundSchema` instead. */
    GetSessionClipsResponse$.outboundSchema = exports.GetSessionClipsResponse$outboundSchema;
})(GetSessionClipsResponse$ || (exports.GetSessionClipsResponse$ = GetSessionClipsResponse$ = {}));
function getSessionClipsResponseToJSON(getSessionClipsResponse) {
    return JSON.stringify(exports.GetSessionClipsResponse$outboundSchema.parse(getSessionClipsResponse));
}
function getSessionClipsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetSessionClipsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetSessionClipsResponse' from JSON`);
}
