// livepeer/models/operations/getsigningkeys.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSigningKeysResponse$ = exports.GetSigningKeysResponse$outboundSchema = exports.GetSigningKeysResponse$inboundSchema = void 0;
exports.getSigningKeysResponseToJSON = getSigningKeysResponseToJSON;
exports.getSigningKeysResponseFromJSON = getSigningKeysResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetSigningKeysResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    data: z.array(components.SigningKey$inboundSchema).optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetSigningKeysResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    data: z.array(components.SigningKey$outboundSchema).optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetSigningKeysResponse$;
(function (GetSigningKeysResponse$) {
    /** @deprecated use `GetSigningKeysResponse$inboundSchema` instead. */
    GetSigningKeysResponse$.inboundSchema = exports.GetSigningKeysResponse$inboundSchema;
    /** @deprecated use `GetSigningKeysResponse$outboundSchema` instead. */
    GetSigningKeysResponse$.outboundSchema = exports.GetSigningKeysResponse$outboundSchema;
})(GetSigningKeysResponse$ || (exports.GetSigningKeysResponse$ = GetSigningKeysResponse$ = {}));
function getSigningKeysResponseToJSON(getSigningKeysResponse) {
    return JSON.stringify(exports.GetSigningKeysResponse$outboundSchema.parse(getSigningKeysResponse));
}
function getSigningKeysResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetSigningKeysResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetSigningKeysResponse' from JSON`);
}
