// livepeer/models/operations/getsigningkey.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSigningKeyResponse$ = exports.GetSigningKeyResponse$outboundSchema = exports.GetSigningKeyResponse$inboundSchema = exports.GetSigningKeyRequest$ = exports.GetSigningKeyRequest$outboundSchema = exports.GetSigningKeyRequest$inboundSchema = void 0;
exports.getSigningKeyRequestToJSON = getSigningKeyRequestToJSON;
exports.getSigningKeyRequestFromJSON = getSigningKeyRequestFromJSON;
exports.getSigningKeyResponseToJSON = getSigningKeyResponseToJSON;
exports.getSigningKeyResponseFromJSON = getSigningKeyResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetSigningKeyRequest$inboundSchema = z.object({
    keyId: z.string(),
});
/** @internal */
exports.GetSigningKeyRequest$outboundSchema = z.object({
    keyId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetSigningKeyRequest$;
(function (GetSigningKeyRequest$) {
    /** @deprecated use `GetSigningKeyRequest$inboundSchema` instead. */
    GetSigningKeyRequest$.inboundSchema = exports.GetSigningKeyRequest$inboundSchema;
    /** @deprecated use `GetSigningKeyRequest$outboundSchema` instead. */
    GetSigningKeyRequest$.outboundSchema = exports.GetSigningKeyRequest$outboundSchema;
})(GetSigningKeyRequest$ || (exports.GetSigningKeyRequest$ = GetSigningKeyRequest$ = {}));
function getSigningKeyRequestToJSON(getSigningKeyRequest) {
    return JSON.stringify(exports.GetSigningKeyRequest$outboundSchema.parse(getSigningKeyRequest));
}
function getSigningKeyRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetSigningKeyRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetSigningKeyRequest' from JSON`);
}
/** @internal */
exports.GetSigningKeyResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    "signing-key": components.SigningKey$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
        "signing-key": "signingKey",
    });
});
/** @internal */
exports.GetSigningKeyResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    signingKey: components.SigningKey$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
        signingKey: "signing-key",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetSigningKeyResponse$;
(function (GetSigningKeyResponse$) {
    /** @deprecated use `GetSigningKeyResponse$inboundSchema` instead. */
    GetSigningKeyResponse$.inboundSchema = exports.GetSigningKeyResponse$inboundSchema;
    /** @deprecated use `GetSigningKeyResponse$outboundSchema` instead. */
    GetSigningKeyResponse$.outboundSchema = exports.GetSigningKeyResponse$outboundSchema;
})(GetSigningKeyResponse$ || (exports.GetSigningKeyResponse$ = GetSigningKeyResponse$ = {}));
function getSigningKeyResponseToJSON(getSigningKeyResponse) {
    return JSON.stringify(exports.GetSigningKeyResponse$outboundSchema.parse(getSigningKeyResponse));
}
function getSigningKeyResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetSigningKeyResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetSigningKeyResponse' from JSON`);
}
