// livepeer/models/operations/gettask.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTaskResponse$ = exports.GetTaskResponse$outboundSchema = exports.GetTaskResponse$inboundSchema = exports.GetTaskRequest$ = exports.GetTaskRequest$outboundSchema = exports.GetTaskRequest$inboundSchema = void 0;
exports.getTaskRequestToJSON = getTaskRequestToJSON;
exports.getTaskRequestFromJSON = getTaskRequestFromJSON;
exports.getTaskResponseToJSON = getTaskResponseToJSON;
exports.getTaskResponseFromJSON = getTaskResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetTaskRequest$inboundSchema = z.object({
    taskId: z.string(),
});
/** @internal */
exports.GetTaskRequest$outboundSchema = z.object({
    taskId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetTaskRequest$;
(function (GetTaskRequest$) {
    /** @deprecated use `GetTaskRequest$inboundSchema` instead. */
    GetTaskRequest$.inboundSchema = exports.GetTaskRequest$inboundSchema;
    /** @deprecated use `GetTaskRequest$outboundSchema` instead. */
    GetTaskRequest$.outboundSchema = exports.GetTaskRequest$outboundSchema;
})(GetTaskRequest$ || (exports.GetTaskRequest$ = GetTaskRequest$ = {}));
function getTaskRequestToJSON(getTaskRequest) {
    return JSON.stringify(exports.GetTaskRequest$outboundSchema.parse(getTaskRequest));
}
function getTaskRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetTaskRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTaskRequest' from JSON`);
}
/** @internal */
exports.GetTaskResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    task: components.Task$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetTaskResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    task: components.Task$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetTaskResponse$;
(function (GetTaskResponse$) {
    /** @deprecated use `GetTaskResponse$inboundSchema` instead. */
    GetTaskResponse$.inboundSchema = exports.GetTaskResponse$inboundSchema;
    /** @deprecated use `GetTaskResponse$outboundSchema` instead. */
    GetTaskResponse$.outboundSchema = exports.GetTaskResponse$outboundSchema;
})(GetTaskResponse$ || (exports.GetTaskResponse$ = GetTaskResponse$ = {}));
function getTaskResponseToJSON(getTaskResponse) {
    return JSON.stringify(exports.GetTaskResponse$outboundSchema.parse(getTaskResponse));
}
function getTaskResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetTaskResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTaskResponse' from JSON`);
}
