// livepeer/models/operations/getwebhooks.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWebhooksResponse$ = exports.GetWebhooksResponse$outboundSchema = exports.GetWebhooksResponse$inboundSchema = void 0;
exports.getWebhooksResponseToJSON = getWebhooksResponseToJSON;
exports.getWebhooksResponseFromJSON = getWebhooksResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetWebhooksResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    data: z.array(components.Webhook$inboundSchema).optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetWebhooksResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    data: z.array(components.Webhook$outboundSchema).optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebhooksResponse$;
(function (GetWebhooksResponse$) {
    /** @deprecated use `GetWebhooksResponse$inboundSchema` instead. */
    GetWebhooksResponse$.inboundSchema = exports.GetWebhooksResponse$inboundSchema;
    /** @deprecated use `GetWebhooksResponse$outboundSchema` instead. */
    GetWebhooksResponse$.outboundSchema = exports.GetWebhooksResponse$outboundSchema;
})(GetWebhooksResponse$ || (exports.GetWebhooksResponse$ = GetWebhooksResponse$ = {}));
function getWebhooksResponseToJSON(getWebhooksResponse) {
    return JSON.stringify(exports.GetWebhooksResponse$outboundSchema.parse(getWebhooksResponse));
}
function getWebhooksResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWebhooksResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWebhooksResponse' from JSON`);
}
