// livepeer/models/operations/getwebhooklogs.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWebhookLogsResponse$ = exports.GetWebhookLogsResponse$outboundSchema = exports.GetWebhookLogsResponse$inboundSchema = exports.GetWebhookLogsRequest$ = exports.GetWebhookLogsRequest$outboundSchema = exports.GetWebhookLogsRequest$inboundSchema = void 0;
exports.getWebhookLogsRequestToJSON = getWebhookLogsRequestToJSON;
exports.getWebhookLogsRequestFromJSON = getWebhookLogsRequestFromJSON;
exports.getWebhookLogsResponseToJSON = getWebhookLogsResponseToJSON;
exports.getWebhookLogsResponseFromJSON = getWebhookLogsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetWebhookLogsRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.GetWebhookLogsRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebhookLogsRequest$;
(function (GetWebhookLogsRequest$) {
    /** @deprecated use `GetWebhookLogsRequest$inboundSchema` instead. */
    GetWebhookLogsRequest$.inboundSchema = exports.GetWebhookLogsRequest$inboundSchema;
    /** @deprecated use `GetWebhookLogsRequest$outboundSchema` instead. */
    GetWebhookLogsRequest$.outboundSchema = exports.GetWebhookLogsRequest$outboundSchema;
})(GetWebhookLogsRequest$ || (exports.GetWebhookLogsRequest$ = GetWebhookLogsRequest$ = {}));
function getWebhookLogsRequestToJSON(getWebhookLogsRequest) {
    return JSON.stringify(exports.GetWebhookLogsRequest$outboundSchema.parse(getWebhookLogsRequest));
}
function getWebhookLogsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWebhookLogsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWebhookLogsRequest' from JSON`);
}
/** @internal */
exports.GetWebhookLogsResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    data: z.array(components.WebhookLog$inboundSchema).optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetWebhookLogsResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    data: z.array(components.WebhookLog$outboundSchema).optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebhookLogsResponse$;
(function (GetWebhookLogsResponse$) {
    /** @deprecated use `GetWebhookLogsResponse$inboundSchema` instead. */
    GetWebhookLogsResponse$.inboundSchema = exports.GetWebhookLogsResponse$inboundSchema;
    /** @deprecated use `GetWebhookLogsResponse$outboundSchema` instead. */
    GetWebhookLogsResponse$.outboundSchema = exports.GetWebhookLogsResponse$outboundSchema;
})(GetWebhookLogsResponse$ || (exports.GetWebhookLogsResponse$ = GetWebhookLogsResponse$ = {}));
function getWebhookLogsResponseToJSON(getWebhookLogsResponse) {
    return JSON.stringify(exports.GetWebhookLogsResponse$outboundSchema.parse(getWebhookLogsResponse));
}
function getWebhookLogsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWebhookLogsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWebhookLogsResponse' from JSON`);
}
