// livepeer/models/operations/getwebhooklog.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWebhookLogResponse$ = exports.GetWebhookLogResponse$outboundSchema = exports.GetWebhookLogResponse$inboundSchema = exports.GetWebhookLogRequest$ = exports.GetWebhookLogRequest$outboundSchema = exports.GetWebhookLogRequest$inboundSchema = void 0;
exports.getWebhookLogRequestToJSON = getWebhookLogRequestToJSON;
exports.getWebhookLogRequestFromJSON = getWebhookLogRequestFromJSON;
exports.getWebhookLogResponseToJSON = getWebhookLogResponseToJSON;
exports.getWebhookLogResponseFromJSON = getWebhookLogResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetWebhookLogRequest$inboundSchema = z.object({
    id: z.string(),
    logId: z.string(),
});
/** @internal */
exports.GetWebhookLogRequest$outboundSchema = z.object({
    id: z.string(),
    logId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebhookLogRequest$;
(function (GetWebhookLogRequest$) {
    /** @deprecated use `GetWebhookLogRequest$inboundSchema` instead. */
    GetWebhookLogRequest$.inboundSchema = exports.GetWebhookLogRequest$inboundSchema;
    /** @deprecated use `GetWebhookLogRequest$outboundSchema` instead. */
    GetWebhookLogRequest$.outboundSchema = exports.GetWebhookLogRequest$outboundSchema;
})(GetWebhookLogRequest$ || (exports.GetWebhookLogRequest$ = GetWebhookLogRequest$ = {}));
function getWebhookLogRequestToJSON(getWebhookLogRequest) {
    return JSON.stringify(exports.GetWebhookLogRequest$outboundSchema.parse(getWebhookLogRequest));
}
function getWebhookLogRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWebhookLogRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWebhookLogRequest' from JSON`);
}
/** @internal */
exports.GetWebhookLogResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    "webhook-log": components.WebhookLog$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
        "webhook-log": "webhookLog",
    });
});
/** @internal */
exports.GetWebhookLogResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    webhookLog: components.WebhookLog$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
        webhookLog: "webhook-log",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebhookLogResponse$;
(function (GetWebhookLogResponse$) {
    /** @deprecated use `GetWebhookLogResponse$inboundSchema` instead. */
    GetWebhookLogResponse$.inboundSchema = exports.GetWebhookLogResponse$inboundSchema;
    /** @deprecated use `GetWebhookLogResponse$outboundSchema` instead. */
    GetWebhookLogResponse$.outboundSchema = exports.GetWebhookLogResponse$outboundSchema;
})(GetWebhookLogResponse$ || (exports.GetWebhookLogResponse$ = GetWebhookLogResponse$ = {}));
function getWebhookLogResponseToJSON(getWebhookLogResponse) {
    return JSON.stringify(exports.GetWebhookLogResponse$outboundSchema.parse(getWebhookLogResponse));
}
function getWebhookLogResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWebhookLogResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWebhookLogResponse' from JSON`);
}
