// livepeer/models/operations/getwebhook.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWebhookResponse$ = exports.GetWebhookResponse$outboundSchema = exports.GetWebhookResponse$inboundSchema = exports.GetWebhookRequest$ = exports.GetWebhookRequest$outboundSchema = exports.GetWebhookRequest$inboundSchema = void 0;
exports.getWebhookRequestToJSON = getWebhookRequestToJSON;
exports.getWebhookRequestFromJSON = getWebhookRequestFromJSON;
exports.getWebhookResponseToJSON = getWebhookResponseToJSON;
exports.getWebhookResponseFromJSON = getWebhookResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetWebhookRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.GetWebhookRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebhookRequest$;
(function (GetWebhookRequest$) {
    /** @deprecated use `GetWebhookRequest$inboundSchema` instead. */
    GetWebhookRequest$.inboundSchema = exports.GetWebhookRequest$inboundSchema;
    /** @deprecated use `GetWebhookRequest$outboundSchema` instead. */
    GetWebhookRequest$.outboundSchema = exports.GetWebhookRequest$outboundSchema;
})(GetWebhookRequest$ || (exports.GetWebhookRequest$ = GetWebhookRequest$ = {}));
function getWebhookRequestToJSON(getWebhookRequest) {
    return JSON.stringify(exports.GetWebhookRequest$outboundSchema.parse(getWebhookRequest));
}
function getWebhookRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWebhookRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWebhookRequest' from JSON`);
}
/** @internal */
exports.GetWebhookResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    webhook: components.Webhook$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetWebhookResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    webhook: components.Webhook$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebhookResponse$;
(function (GetWebhookResponse$) {
    /** @deprecated use `GetWebhookResponse$inboundSchema` instead. */
    GetWebhookResponse$.inboundSchema = exports.GetWebhookResponse$inboundSchema;
    /** @deprecated use `GetWebhookResponse$outboundSchema` instead. */
    GetWebhookResponse$.outboundSchema = exports.GetWebhookResponse$outboundSchema;
})(GetWebhookResponse$ || (exports.GetWebhookResponse$ = GetWebhookResponse$ = {}));
function getWebhookResponseToJSON(getWebhookResponse) {
    return JSON.stringify(exports.GetWebhookResponse$outboundSchema.parse(getWebhookResponse));
}
function getWebhookResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetWebhookResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetWebhookResponse' from JSON`);
}
