// livepeer/models/operations/removemultistreamtarget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoveMultistreamTargetResponse$ = exports.RemoveMultistreamTargetResponse$outboundSchema = exports.RemoveMultistreamTargetResponse$inboundSchema = exports.RemoveMultistreamTargetRequest$ = exports.RemoveMultistreamTargetRequest$outboundSchema = exports.RemoveMultistreamTargetRequest$inboundSchema = void 0;
exports.removeMultistreamTargetRequestToJSON = removeMultistreamTargetRequestToJSON;
exports.removeMultistreamTargetRequestFromJSON = removeMultistreamTargetRequestFromJSON;
exports.removeMultistreamTargetResponseToJSON = removeMultistreamTargetResponseToJSON;
exports.removeMultistreamTargetResponseFromJSON = removeMultistreamTargetResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.RemoveMultistreamTargetRequest$inboundSchema = z.object({
    id: z.string(),
    targetId: z.string(),
});
/** @internal */
exports.RemoveMultistreamTargetRequest$outboundSchema = z.object({
    id: z.string(),
    targetId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RemoveMultistreamTargetRequest$;
(function (RemoveMultistreamTargetRequest$) {
    /** @deprecated use `RemoveMultistreamTargetRequest$inboundSchema` instead. */
    RemoveMultistreamTargetRequest$.inboundSchema = exports.RemoveMultistreamTargetRequest$inboundSchema;
    /** @deprecated use `RemoveMultistreamTargetRequest$outboundSchema` instead. */
    RemoveMultistreamTargetRequest$.outboundSchema = exports.RemoveMultistreamTargetRequest$outboundSchema;
})(RemoveMultistreamTargetRequest$ || (exports.RemoveMultistreamTargetRequest$ = RemoveMultistreamTargetRequest$ = {}));
function removeMultistreamTargetRequestToJSON(removeMultistreamTargetRequest) {
    return JSON.stringify(exports.RemoveMultistreamTargetRequest$outboundSchema.parse(removeMultistreamTargetRequest));
}
function removeMultistreamTargetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RemoveMultistreamTargetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveMultistreamTargetRequest' from JSON`);
}
/** @internal */
exports.RemoveMultistreamTargetResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.RemoveMultistreamTargetResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RemoveMultistreamTargetResponse$;
(function (RemoveMultistreamTargetResponse$) {
    /** @deprecated use `RemoveMultistreamTargetResponse$inboundSchema` instead. */
    RemoveMultistreamTargetResponse$.inboundSchema = exports.RemoveMultistreamTargetResponse$inboundSchema;
    /** @deprecated use `RemoveMultistreamTargetResponse$outboundSchema` instead. */
    RemoveMultistreamTargetResponse$.outboundSchema = exports.RemoveMultistreamTargetResponse$outboundSchema;
})(RemoveMultistreamTargetResponse$ || (exports.RemoveMultistreamTargetResponse$ = RemoveMultistreamTargetResponse$ = {}));
function removeMultistreamTargetResponseToJSON(removeMultistreamTargetResponse) {
    return JSON.stringify(exports.RemoveMultistreamTargetResponse$outboundSchema.parse(removeMultistreamTargetResponse));
}
function removeMultistreamTargetResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RemoveMultistreamTargetResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveMultistreamTargetResponse' from JSON`);
}
