// livepeer/models/operations/startpullstream.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StartPullStreamResponse$ = exports.StartPullStreamResponse$outboundSchema = exports.StartPullStreamResponse$inboundSchema = exports.StartPullStreamRequest$ = exports.StartPullStreamRequest$outboundSchema = exports.StartPullStreamRequest$inboundSchema = void 0;
exports.startPullStreamRequestToJSON = startPullStreamRequestToJSON;
exports.startPullStreamRequestFromJSON = startPullStreamRequestFromJSON;
exports.startPullStreamResponseToJSON = startPullStreamResponseToJSON;
exports.startPullStreamResponseFromJSON = startPullStreamResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.StartPullStreamRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.StartPullStreamRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StartPullStreamRequest$;
(function (StartPullStreamRequest$) {
    /** @deprecated use `StartPullStreamRequest$inboundSchema` instead. */
    StartPullStreamRequest$.inboundSchema = exports.StartPullStreamRequest$inboundSchema;
    /** @deprecated use `StartPullStreamRequest$outboundSchema` instead. */
    StartPullStreamRequest$.outboundSchema = exports.StartPullStreamRequest$outboundSchema;
})(StartPullStreamRequest$ || (exports.StartPullStreamRequest$ = StartPullStreamRequest$ = {}));
function startPullStreamRequestToJSON(startPullStreamRequest) {
    return JSON.stringify(exports.StartPullStreamRequest$outboundSchema.parse(startPullStreamRequest));
}
function startPullStreamRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StartPullStreamRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StartPullStreamRequest' from JSON`);
}
/** @internal */
exports.StartPullStreamResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.StartPullStreamResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StartPullStreamResponse$;
(function (StartPullStreamResponse$) {
    /** @deprecated use `StartPullStreamResponse$inboundSchema` instead. */
    StartPullStreamResponse$.inboundSchema = exports.StartPullStreamResponse$inboundSchema;
    /** @deprecated use `StartPullStreamResponse$outboundSchema` instead. */
    StartPullStreamResponse$.outboundSchema = exports.StartPullStreamResponse$outboundSchema;
})(StartPullStreamResponse$ || (exports.StartPullStreamResponse$ = StartPullStreamResponse$ = {}));
function startPullStreamResponseToJSON(startPullStreamResponse) {
    return JSON.stringify(exports.StartPullStreamResponse$outboundSchema.parse(startPullStreamResponse));
}
function startPullStreamResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StartPullStreamResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StartPullStreamResponse' from JSON`);
}
