// livepeer/models/operations/startroomegress.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StartRoomEgressResponse$ = exports.StartRoomEgressResponse$outboundSchema = exports.StartRoomEgressResponse$inboundSchema = exports.StartRoomEgressRequest$ = exports.StartRoomEgressRequest$outboundSchema = exports.StartRoomEgressRequest$inboundSchema = void 0;
exports.startRoomEgressRequestToJSON = startRoomEgressRequestToJSON;
exports.startRoomEgressRequestFromJSON = startRoomEgressRequestFromJSON;
exports.startRoomEgressResponseToJSON = startRoomEgressResponseToJSON;
exports.startRoomEgressResponseFromJSON = startRoomEgressResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.StartRoomEgressRequest$inboundSchema = z.object({
    id: z.string(),
    "room-egress-payload": components.RoomEgressPayload$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "room-egress-payload": "roomEgressPayload",
    });
});
/** @internal */
exports.StartRoomEgressRequest$outboundSchema = z.object({
    id: z.string(),
    roomEgressPayload: components.RoomEgressPayload$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        roomEgressPayload: "room-egress-payload",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StartRoomEgressRequest$;
(function (StartRoomEgressRequest$) {
    /** @deprecated use `StartRoomEgressRequest$inboundSchema` instead. */
    StartRoomEgressRequest$.inboundSchema = exports.StartRoomEgressRequest$inboundSchema;
    /** @deprecated use `StartRoomEgressRequest$outboundSchema` instead. */
    StartRoomEgressRequest$.outboundSchema = exports.StartRoomEgressRequest$outboundSchema;
})(StartRoomEgressRequest$ || (exports.StartRoomEgressRequest$ = StartRoomEgressRequest$ = {}));
function startRoomEgressRequestToJSON(startRoomEgressRequest) {
    return JSON.stringify(exports.StartRoomEgressRequest$outboundSchema.parse(startRoomEgressRequest));
}
function startRoomEgressRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StartRoomEgressRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StartRoomEgressRequest' from JSON`);
}
/** @internal */
exports.StartRoomEgressResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.StartRoomEgressResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StartRoomEgressResponse$;
(function (StartRoomEgressResponse$) {
    /** @deprecated use `StartRoomEgressResponse$inboundSchema` instead. */
    StartRoomEgressResponse$.inboundSchema = exports.StartRoomEgressResponse$inboundSchema;
    /** @deprecated use `StartRoomEgressResponse$outboundSchema` instead. */
    StartRoomEgressResponse$.outboundSchema = exports.StartRoomEgressResponse$outboundSchema;
})(StartRoomEgressResponse$ || (exports.StartRoomEgressResponse$ = StartRoomEgressResponse$ = {}));
function startRoomEgressResponseToJSON(startRoomEgressResponse) {
    return JSON.stringify(exports.StartRoomEgressResponse$outboundSchema.parse(startRoomEgressResponse));
}
function startRoomEgressResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StartRoomEgressResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StartRoomEgressResponse' from JSON`);
}
