// livepeer/models/operations/stoproomegress.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StopRoomEgressResponse$ = exports.StopRoomEgressResponse$outboundSchema = exports.StopRoomEgressResponse$inboundSchema = exports.StopRoomEgressRequest$ = exports.StopRoomEgressRequest$outboundSchema = exports.StopRoomEgressRequest$inboundSchema = void 0;
exports.stopRoomEgressRequestToJSON = stopRoomEgressRequestToJSON;
exports.stopRoomEgressRequestFromJSON = stopRoomEgressRequestFromJSON;
exports.stopRoomEgressResponseToJSON = stopRoomEgressResponseToJSON;
exports.stopRoomEgressResponseFromJSON = stopRoomEgressResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.StopRoomEgressRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.StopRoomEgressRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StopRoomEgressRequest$;
(function (StopRoomEgressRequest$) {
    /** @deprecated use `StopRoomEgressRequest$inboundSchema` instead. */
    StopRoomEgressRequest$.inboundSchema = exports.StopRoomEgressRequest$inboundSchema;
    /** @deprecated use `StopRoomEgressRequest$outboundSchema` instead. */
    StopRoomEgressRequest$.outboundSchema = exports.StopRoomEgressRequest$outboundSchema;
})(StopRoomEgressRequest$ || (exports.StopRoomEgressRequest$ = StopRoomEgressRequest$ = {}));
function stopRoomEgressRequestToJSON(stopRoomEgressRequest) {
    return JSON.stringify(exports.StopRoomEgressRequest$outboundSchema.parse(stopRoomEgressRequest));
}
function stopRoomEgressRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StopRoomEgressRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StopRoomEgressRequest' from JSON`);
}
/** @internal */
exports.StopRoomEgressResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.StopRoomEgressResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StopRoomEgressResponse$;
(function (StopRoomEgressResponse$) {
    /** @deprecated use `StopRoomEgressResponse$inboundSchema` instead. */
    StopRoomEgressResponse$.inboundSchema = exports.StopRoomEgressResponse$inboundSchema;
    /** @deprecated use `StopRoomEgressResponse$outboundSchema` instead. */
    StopRoomEgressResponse$.outboundSchema = exports.StopRoomEgressResponse$outboundSchema;
})(StopRoomEgressResponse$ || (exports.StopRoomEgressResponse$ = StopRoomEgressResponse$ = {}));
function stopRoomEgressResponseToJSON(stopRoomEgressResponse) {
    return JSON.stringify(exports.StopRoomEgressResponse$outboundSchema.parse(stopRoomEgressResponse));
}
function stopRoomEgressResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StopRoomEgressResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StopRoomEgressResponse' from JSON`);
}
