// livepeer/models/operations/terminatestream.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminateStreamResponse$ = exports.TerminateStreamResponse$outboundSchema = exports.TerminateStreamResponse$inboundSchema = exports.TerminateStreamRequest$ = exports.TerminateStreamRequest$outboundSchema = exports.TerminateStreamRequest$inboundSchema = void 0;
exports.terminateStreamRequestToJSON = terminateStreamRequestToJSON;
exports.terminateStreamRequestFromJSON = terminateStreamRequestFromJSON;
exports.terminateStreamResponseToJSON = terminateStreamResponseToJSON;
exports.terminateStreamResponseFromJSON = terminateStreamResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.TerminateStreamRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.TerminateStreamRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TerminateStreamRequest$;
(function (TerminateStreamRequest$) {
    /** @deprecated use `TerminateStreamRequest$inboundSchema` instead. */
    TerminateStreamRequest$.inboundSchema = exports.TerminateStreamRequest$inboundSchema;
    /** @deprecated use `TerminateStreamRequest$outboundSchema` instead. */
    TerminateStreamRequest$.outboundSchema = exports.TerminateStreamRequest$outboundSchema;
})(TerminateStreamRequest$ || (exports.TerminateStreamRequest$ = TerminateStreamRequest$ = {}));
function terminateStreamRequestToJSON(terminateStreamRequest) {
    return JSON.stringify(exports.TerminateStreamRequest$outboundSchema.parse(terminateStreamRequest));
}
function terminateStreamRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TerminateStreamRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TerminateStreamRequest' from JSON`);
}
/** @internal */
exports.TerminateStreamResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.TerminateStreamResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TerminateStreamResponse$;
(function (TerminateStreamResponse$) {
    /** @deprecated use `TerminateStreamResponse$inboundSchema` instead. */
    TerminateStreamResponse$.inboundSchema = exports.TerminateStreamResponse$inboundSchema;
    /** @deprecated use `TerminateStreamResponse$outboundSchema` instead. */
    TerminateStreamResponse$.outboundSchema = exports.TerminateStreamResponse$outboundSchema;
})(TerminateStreamResponse$ || (exports.TerminateStreamResponse$ = TerminateStreamResponse$ = {}));
function terminateStreamResponseToJSON(terminateStreamResponse) {
    return JSON.stringify(exports.TerminateStreamResponse$outboundSchema.parse(terminateStreamResponse));
}
function terminateStreamResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TerminateStreamResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TerminateStreamResponse' from JSON`);
}
