// livepeer/models/operations/updatemultistreamtarget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateMultistreamTargetResponse$ = exports.UpdateMultistreamTargetResponse$outboundSchema = exports.UpdateMultistreamTargetResponse$inboundSchema = exports.UpdateMultistreamTargetRequest$ = exports.UpdateMultistreamTargetRequest$outboundSchema = exports.UpdateMultistreamTargetRequest$inboundSchema = void 0;
exports.updateMultistreamTargetRequestToJSON = updateMultistreamTargetRequestToJSON;
exports.updateMultistreamTargetRequestFromJSON = updateMultistreamTargetRequestFromJSON;
exports.updateMultistreamTargetResponseToJSON = updateMultistreamTargetResponseToJSON;
exports.updateMultistreamTargetResponseFromJSON = updateMultistreamTargetResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UpdateMultistreamTargetRequest$inboundSchema = z.object({
    id: z.string(),
    "multistream-target-patch-payload": components.MultistreamTargetPatchPayload$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "multistream-target-patch-payload": "multistreamTargetPatchPayload",
    });
});
/** @internal */
exports.UpdateMultistreamTargetRequest$outboundSchema = z.object({
    id: z.string(),
    multistreamTargetPatchPayload: components.MultistreamTargetPatchPayload$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        multistreamTargetPatchPayload: "multistream-target-patch-payload",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateMultistreamTargetRequest$;
(function (UpdateMultistreamTargetRequest$) {
    /** @deprecated use `UpdateMultistreamTargetRequest$inboundSchema` instead. */
    UpdateMultistreamTargetRequest$.inboundSchema = exports.UpdateMultistreamTargetRequest$inboundSchema;
    /** @deprecated use `UpdateMultistreamTargetRequest$outboundSchema` instead. */
    UpdateMultistreamTargetRequest$.outboundSchema = exports.UpdateMultistreamTargetRequest$outboundSchema;
})(UpdateMultistreamTargetRequest$ || (exports.UpdateMultistreamTargetRequest$ = UpdateMultistreamTargetRequest$ = {}));
function updateMultistreamTargetRequestToJSON(updateMultistreamTargetRequest) {
    return JSON.stringify(exports.UpdateMultistreamTargetRequest$outboundSchema.parse(updateMultistreamTargetRequest));
}
function updateMultistreamTargetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateMultistreamTargetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateMultistreamTargetRequest' from JSON`);
}
/** @internal */
exports.UpdateMultistreamTargetResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.UpdateMultistreamTargetResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateMultistreamTargetResponse$;
(function (UpdateMultistreamTargetResponse$) {
    /** @deprecated use `UpdateMultistreamTargetResponse$inboundSchema` instead. */
    UpdateMultistreamTargetResponse$.inboundSchema = exports.UpdateMultistreamTargetResponse$inboundSchema;
    /** @deprecated use `UpdateMultistreamTargetResponse$outboundSchema` instead. */
    UpdateMultistreamTargetResponse$.outboundSchema = exports.UpdateMultistreamTargetResponse$outboundSchema;
})(UpdateMultistreamTargetResponse$ || (exports.UpdateMultistreamTargetResponse$ = UpdateMultistreamTargetResponse$ = {}));
function updateMultistreamTargetResponseToJSON(updateMultistreamTargetResponse) {
    return JSON.stringify(exports.UpdateMultistreamTargetResponse$outboundSchema.parse(updateMultistreamTargetResponse));
}
function updateMultistreamTargetResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateMultistreamTargetResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateMultistreamTargetResponse' from JSON`);
}
