// livepeer/models/operations/updateasset.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAssetResponse$ = exports.UpdateAssetResponse$outboundSchema = exports.UpdateAssetResponse$inboundSchema = exports.UpdateAssetRequest$ = exports.UpdateAssetRequest$outboundSchema = exports.UpdateAssetRequest$inboundSchema = void 0;
exports.updateAssetRequestToJSON = updateAssetRequestToJSON;
exports.updateAssetRequestFromJSON = updateAssetRequestFromJSON;
exports.updateAssetResponseToJSON = updateAssetResponseToJSON;
exports.updateAssetResponseFromJSON = updateAssetResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UpdateAssetRequest$inboundSchema = z.object({
    assetId: z.string(),
    "asset-patch-payload": components.AssetPatchPayload$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "asset-patch-payload": "assetPatchPayload",
    });
});
/** @internal */
exports.UpdateAssetRequest$outboundSchema = z.object({
    assetId: z.string(),
    assetPatchPayload: components.AssetPatchPayload$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        assetPatchPayload: "asset-patch-payload",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateAssetRequest$;
(function (UpdateAssetRequest$) {
    /** @deprecated use `UpdateAssetRequest$inboundSchema` instead. */
    UpdateAssetRequest$.inboundSchema = exports.UpdateAssetRequest$inboundSchema;
    /** @deprecated use `UpdateAssetRequest$outboundSchema` instead. */
    UpdateAssetRequest$.outboundSchema = exports.UpdateAssetRequest$outboundSchema;
})(UpdateAssetRequest$ || (exports.UpdateAssetRequest$ = UpdateAssetRequest$ = {}));
function updateAssetRequestToJSON(updateAssetRequest) {
    return JSON.stringify(exports.UpdateAssetRequest$outboundSchema.parse(updateAssetRequest));
}
function updateAssetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateAssetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateAssetRequest' from JSON`);
}
/** @internal */
exports.UpdateAssetResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    asset: components.Asset$inboundSchema.optional(),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.UpdateAssetResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    asset: components.Asset$outboundSchema.optional(),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateAssetResponse$;
(function (UpdateAssetResponse$) {
    /** @deprecated use `UpdateAssetResponse$inboundSchema` instead. */
    UpdateAssetResponse$.inboundSchema = exports.UpdateAssetResponse$inboundSchema;
    /** @deprecated use `UpdateAssetResponse$outboundSchema` instead. */
    UpdateAssetResponse$.outboundSchema = exports.UpdateAssetResponse$outboundSchema;
})(UpdateAssetResponse$ || (exports.UpdateAssetResponse$ = UpdateAssetResponse$ = {}));
function updateAssetResponseToJSON(updateAssetResponse) {
    return JSON.stringify(exports.UpdateAssetResponse$outboundSchema.parse(updateAssetResponse));
}
function updateAssetResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateAssetResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateAssetResponse' from JSON`);
}
