// livepeer/models/operations/updatestream.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateStreamResponse$ = exports.UpdateStreamResponse$outboundSchema = exports.UpdateStreamResponse$inboundSchema = exports.UpdateStreamRequest$ = exports.UpdateStreamRequest$outboundSchema = exports.UpdateStreamRequest$inboundSchema = void 0;
exports.updateStreamRequestToJSON = updateStreamRequestToJSON;
exports.updateStreamRequestFromJSON = updateStreamRequestFromJSON;
exports.updateStreamResponseToJSON = updateStreamResponseToJSON;
exports.updateStreamResponseFromJSON = updateStreamResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UpdateStreamRequest$inboundSchema = z.object({
    id: z.string(),
    "stream-patch-payload": components.StreamPatchPayload$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "stream-patch-payload": "streamPatchPayload",
    });
});
/** @internal */
exports.UpdateStreamRequest$outboundSchema = z.object({
    id: z.string(),
    streamPatchPayload: components.StreamPatchPayload$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        streamPatchPayload: "stream-patch-payload",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateStreamRequest$;
(function (UpdateStreamRequest$) {
    /** @deprecated use `UpdateStreamRequest$inboundSchema` instead. */
    UpdateStreamRequest$.inboundSchema = exports.UpdateStreamRequest$inboundSchema;
    /** @deprecated use `UpdateStreamRequest$outboundSchema` instead. */
    UpdateStreamRequest$.outboundSchema = exports.UpdateStreamRequest$outboundSchema;
})(UpdateStreamRequest$ || (exports.UpdateStreamRequest$ = UpdateStreamRequest$ = {}));
function updateStreamRequestToJSON(updateStreamRequest) {
    return JSON.stringify(exports.UpdateStreamRequest$outboundSchema.parse(updateStreamRequest));
}
function updateStreamRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateStreamRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateStreamRequest' from JSON`);
}
/** @internal */
exports.UpdateStreamResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    error: components.ErrorT$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.UpdateStreamResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    error: components.ErrorT$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateStreamResponse$;
(function (UpdateStreamResponse$) {
    /** @deprecated use `UpdateStreamResponse$inboundSchema` instead. */
    UpdateStreamResponse$.inboundSchema = exports.UpdateStreamResponse$inboundSchema;
    /** @deprecated use `UpdateStreamResponse$outboundSchema` instead. */
    UpdateStreamResponse$.outboundSchema = exports.UpdateStreamResponse$outboundSchema;
})(UpdateStreamResponse$ || (exports.UpdateStreamResponse$ = UpdateStreamResponse$ = {}));
function updateStreamResponseToJSON(updateStreamResponse) {
    return JSON.stringify(exports.UpdateStreamResponse$outboundSchema.parse(updateStreamResponse));
}
function updateStreamResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateStreamResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateStreamResponse' from JSON`);
}
