// livepeer/models/errors/gensegmentanything2.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenSegmentAnything2ResponseBody$ = exports.GenSegmentAnything2ResponseBody$outboundSchema = exports.GenSegmentAnything2ResponseBody$inboundSchema = exports.GenSegmentAnything2GenerateResponseBody$ = exports.GenSegmentAnything2GenerateResponseBody$outboundSchema = exports.GenSegmentAnything2GenerateResponseBody$inboundSchema = exports.GenSegmentAnything2GenerateResponseResponseBody$ = exports.GenSegmentAnything2GenerateResponseResponseBody$outboundSchema = exports.GenSegmentAnything2GenerateResponseResponseBody$inboundSchema = exports.GenSegmentAnything2GenerateResponse500ResponseBody$ = exports.GenSegmentAnything2GenerateResponse500ResponseBody$outboundSchema = exports.GenSegmentAnything2GenerateResponse500ResponseBody$inboundSchema = void 0;
exports.genSegmentAnything2GenerateResponse500ResponseBodyToJSON = genSegmentAnything2GenerateResponse500ResponseBodyToJSON;
exports.genSegmentAnything2GenerateResponse500ResponseBodyFromJSON = genSegmentAnything2GenerateResponse500ResponseBodyFromJSON;
exports.genSegmentAnything2GenerateResponseResponseBodyToJSON = genSegmentAnything2GenerateResponseResponseBodyToJSON;
exports.genSegmentAnything2GenerateResponseResponseBodyFromJSON = genSegmentAnything2GenerateResponseResponseBodyFromJSON;
exports.genSegmentAnything2GenerateResponseBodyToJSON = genSegmentAnything2GenerateResponseBodyToJSON;
exports.genSegmentAnything2GenerateResponseBodyFromJSON = genSegmentAnything2GenerateResponseBodyFromJSON;
exports.genSegmentAnything2ResponseBodyToJSON = genSegmentAnything2ResponseBodyToJSON;
exports.genSegmentAnything2ResponseBodyFromJSON = genSegmentAnything2ResponseBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const httperror_js_1 = require("./httperror.js");
const httpvalidationerror_js_1 = require("./httpvalidationerror.js");
const studioapierror_js_1 = require("./studioapierror.js");
/** @internal */
exports.GenSegmentAnything2GenerateResponse500ResponseBody$inboundSchema = z.union([httperror_js_1.HTTPError$inboundSchema, studioapierror_js_1.StudioApiError$inboundSchema]);
/** @internal */
exports.GenSegmentAnything2GenerateResponse500ResponseBody$outboundSchema = z.union([httperror_js_1.HTTPError$outboundSchema, studioapierror_js_1.StudioApiError$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenSegmentAnything2GenerateResponse500ResponseBody$;
(function (GenSegmentAnything2GenerateResponse500ResponseBody$) {
    /** @deprecated use `GenSegmentAnything2GenerateResponse500ResponseBody$inboundSchema` instead. */
    GenSegmentAnything2GenerateResponse500ResponseBody$.inboundSchema = exports.GenSegmentAnything2GenerateResponse500ResponseBody$inboundSchema;
    /** @deprecated use `GenSegmentAnything2GenerateResponse500ResponseBody$outboundSchema` instead. */
    GenSegmentAnything2GenerateResponse500ResponseBody$.outboundSchema = exports.GenSegmentAnything2GenerateResponse500ResponseBody$outboundSchema;
})(GenSegmentAnything2GenerateResponse500ResponseBody$ || (exports.GenSegmentAnything2GenerateResponse500ResponseBody$ = GenSegmentAnything2GenerateResponse500ResponseBody$ = {}));
function genSegmentAnything2GenerateResponse500ResponseBodyToJSON(genSegmentAnything2GenerateResponse500ResponseBody) {
    return JSON.stringify(exports.GenSegmentAnything2GenerateResponse500ResponseBody$outboundSchema.parse(genSegmentAnything2GenerateResponse500ResponseBody));
}
function genSegmentAnything2GenerateResponse500ResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenSegmentAnything2GenerateResponse500ResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenSegmentAnything2GenerateResponse500ResponseBody' from JSON`);
}
/** @internal */
exports.GenSegmentAnything2GenerateResponseResponseBody$inboundSchema = z.union([
    httpvalidationerror_js_1.HTTPValidationError$inboundSchema,
    studioapierror_js_1.StudioApiError$inboundSchema,
]);
/** @internal */
exports.GenSegmentAnything2GenerateResponseResponseBody$outboundSchema = z.union([
    httpvalidationerror_js_1.HTTPValidationError$outboundSchema,
    studioapierror_js_1.StudioApiError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenSegmentAnything2GenerateResponseResponseBody$;
(function (GenSegmentAnything2GenerateResponseResponseBody$) {
    /** @deprecated use `GenSegmentAnything2GenerateResponseResponseBody$inboundSchema` instead. */
    GenSegmentAnything2GenerateResponseResponseBody$.inboundSchema = exports.GenSegmentAnything2GenerateResponseResponseBody$inboundSchema;
    /** @deprecated use `GenSegmentAnything2GenerateResponseResponseBody$outboundSchema` instead. */
    GenSegmentAnything2GenerateResponseResponseBody$.outboundSchema = exports.GenSegmentAnything2GenerateResponseResponseBody$outboundSchema;
})(GenSegmentAnything2GenerateResponseResponseBody$ || (exports.GenSegmentAnything2GenerateResponseResponseBody$ = GenSegmentAnything2GenerateResponseResponseBody$ = {}));
function genSegmentAnything2GenerateResponseResponseBodyToJSON(genSegmentAnything2GenerateResponseResponseBody) {
    return JSON.stringify(exports.GenSegmentAnything2GenerateResponseResponseBody$outboundSchema.parse(genSegmentAnything2GenerateResponseResponseBody));
}
function genSegmentAnything2GenerateResponseResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenSegmentAnything2GenerateResponseResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenSegmentAnything2GenerateResponseResponseBody' from JSON`);
}
/** @internal */
exports.GenSegmentAnything2GenerateResponseBody$inboundSchema = z.union([httperror_js_1.HTTPError$inboundSchema, studioapierror_js_1.StudioApiError$inboundSchema]);
/** @internal */
exports.GenSegmentAnything2GenerateResponseBody$outboundSchema = z.union([httperror_js_1.HTTPError$outboundSchema, studioapierror_js_1.StudioApiError$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenSegmentAnything2GenerateResponseBody$;
(function (GenSegmentAnything2GenerateResponseBody$) {
    /** @deprecated use `GenSegmentAnything2GenerateResponseBody$inboundSchema` instead. */
    GenSegmentAnything2GenerateResponseBody$.inboundSchema = exports.GenSegmentAnything2GenerateResponseBody$inboundSchema;
    /** @deprecated use `GenSegmentAnything2GenerateResponseBody$outboundSchema` instead. */
    GenSegmentAnything2GenerateResponseBody$.outboundSchema = exports.GenSegmentAnything2GenerateResponseBody$outboundSchema;
})(GenSegmentAnything2GenerateResponseBody$ || (exports.GenSegmentAnything2GenerateResponseBody$ = GenSegmentAnything2GenerateResponseBody$ = {}));
function genSegmentAnything2GenerateResponseBodyToJSON(genSegmentAnything2GenerateResponseBody) {
    return JSON.stringify(exports.GenSegmentAnything2GenerateResponseBody$outboundSchema.parse(genSegmentAnything2GenerateResponseBody));
}
function genSegmentAnything2GenerateResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenSegmentAnything2GenerateResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenSegmentAnything2GenerateResponseBody' from JSON`);
}
/** @internal */
exports.GenSegmentAnything2ResponseBody$inboundSchema = z.union([httperror_js_1.HTTPError$inboundSchema, studioapierror_js_1.StudioApiError$inboundSchema]);
/** @internal */
exports.GenSegmentAnything2ResponseBody$outboundSchema = z.union([httperror_js_1.HTTPError$outboundSchema, studioapierror_js_1.StudioApiError$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenSegmentAnything2ResponseBody$;
(function (GenSegmentAnything2ResponseBody$) {
    /** @deprecated use `GenSegmentAnything2ResponseBody$inboundSchema` instead. */
    GenSegmentAnything2ResponseBody$.inboundSchema = exports.GenSegmentAnything2ResponseBody$inboundSchema;
    /** @deprecated use `GenSegmentAnything2ResponseBody$outboundSchema` instead. */
    GenSegmentAnything2ResponseBody$.outboundSchema = exports.GenSegmentAnything2ResponseBody$outboundSchema;
})(GenSegmentAnything2ResponseBody$ || (exports.GenSegmentAnything2ResponseBody$ = GenSegmentAnything2ResponseBody$ = {}));
function genSegmentAnything2ResponseBodyToJSON(genSegmentAnything2ResponseBody) {
    return JSON.stringify(exports.GenSegmentAnything2ResponseBody$outboundSchema.parse(genSegmentAnything2ResponseBody));
}
function genSegmentAnything2ResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenSegmentAnything2ResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenSegmentAnything2ResponseBody' from JSON`);
}
