// livepeer/models/errors/genupscale.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenUpscaleResponseBody$ = exports.GenUpscaleResponseBody$outboundSchema = exports.GenUpscaleResponseBody$inboundSchema = exports.GenUpscaleGenerateResponseBody$ = exports.GenUpscaleGenerateResponseBody$outboundSchema = exports.GenUpscaleGenerateResponseBody$inboundSchema = exports.GenUpscaleGenerateResponseResponseBody$ = exports.GenUpscaleGenerateResponseResponseBody$outboundSchema = exports.GenUpscaleGenerateResponseResponseBody$inboundSchema = exports.GenUpscaleGenerateResponse500ResponseBody$ = exports.GenUpscaleGenerateResponse500ResponseBody$outboundSchema = exports.GenUpscaleGenerateResponse500ResponseBody$inboundSchema = void 0;
exports.genUpscaleGenerateResponse500ResponseBodyToJSON = genUpscaleGenerateResponse500ResponseBodyToJSON;
exports.genUpscaleGenerateResponse500ResponseBodyFromJSON = genUpscaleGenerateResponse500ResponseBodyFromJSON;
exports.genUpscaleGenerateResponseResponseBodyToJSON = genUpscaleGenerateResponseResponseBodyToJSON;
exports.genUpscaleGenerateResponseResponseBodyFromJSON = genUpscaleGenerateResponseResponseBodyFromJSON;
exports.genUpscaleGenerateResponseBodyToJSON = genUpscaleGenerateResponseBodyToJSON;
exports.genUpscaleGenerateResponseBodyFromJSON = genUpscaleGenerateResponseBodyFromJSON;
exports.genUpscaleResponseBodyToJSON = genUpscaleResponseBodyToJSON;
exports.genUpscaleResponseBodyFromJSON = genUpscaleResponseBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const httperror_js_1 = require("./httperror.js");
const httpvalidationerror_js_1 = require("./httpvalidationerror.js");
const studioapierror_js_1 = require("./studioapierror.js");
/** @internal */
exports.GenUpscaleGenerateResponse500ResponseBody$inboundSchema = z.union([httperror_js_1.HTTPError$inboundSchema, studioapierror_js_1.StudioApiError$inboundSchema]);
/** @internal */
exports.GenUpscaleGenerateResponse500ResponseBody$outboundSchema = z.union([httperror_js_1.HTTPError$outboundSchema, studioapierror_js_1.StudioApiError$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenUpscaleGenerateResponse500ResponseBody$;
(function (GenUpscaleGenerateResponse500ResponseBody$) {
    /** @deprecated use `GenUpscaleGenerateResponse500ResponseBody$inboundSchema` instead. */
    GenUpscaleGenerateResponse500ResponseBody$.inboundSchema = exports.GenUpscaleGenerateResponse500ResponseBody$inboundSchema;
    /** @deprecated use `GenUpscaleGenerateResponse500ResponseBody$outboundSchema` instead. */
    GenUpscaleGenerateResponse500ResponseBody$.outboundSchema = exports.GenUpscaleGenerateResponse500ResponseBody$outboundSchema;
})(GenUpscaleGenerateResponse500ResponseBody$ || (exports.GenUpscaleGenerateResponse500ResponseBody$ = GenUpscaleGenerateResponse500ResponseBody$ = {}));
function genUpscaleGenerateResponse500ResponseBodyToJSON(genUpscaleGenerateResponse500ResponseBody) {
    return JSON.stringify(exports.GenUpscaleGenerateResponse500ResponseBody$outboundSchema.parse(genUpscaleGenerateResponse500ResponseBody));
}
function genUpscaleGenerateResponse500ResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenUpscaleGenerateResponse500ResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenUpscaleGenerateResponse500ResponseBody' from JSON`);
}
/** @internal */
exports.GenUpscaleGenerateResponseResponseBody$inboundSchema = z.union([httpvalidationerror_js_1.HTTPValidationError$inboundSchema, studioapierror_js_1.StudioApiError$inboundSchema]);
/** @internal */
exports.GenUpscaleGenerateResponseResponseBody$outboundSchema = z.union([
    httpvalidationerror_js_1.HTTPValidationError$outboundSchema,
    studioapierror_js_1.StudioApiError$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenUpscaleGenerateResponseResponseBody$;
(function (GenUpscaleGenerateResponseResponseBody$) {
    /** @deprecated use `GenUpscaleGenerateResponseResponseBody$inboundSchema` instead. */
    GenUpscaleGenerateResponseResponseBody$.inboundSchema = exports.GenUpscaleGenerateResponseResponseBody$inboundSchema;
    /** @deprecated use `GenUpscaleGenerateResponseResponseBody$outboundSchema` instead. */
    GenUpscaleGenerateResponseResponseBody$.outboundSchema = exports.GenUpscaleGenerateResponseResponseBody$outboundSchema;
})(GenUpscaleGenerateResponseResponseBody$ || (exports.GenUpscaleGenerateResponseResponseBody$ = GenUpscaleGenerateResponseResponseBody$ = {}));
function genUpscaleGenerateResponseResponseBodyToJSON(genUpscaleGenerateResponseResponseBody) {
    return JSON.stringify(exports.GenUpscaleGenerateResponseResponseBody$outboundSchema.parse(genUpscaleGenerateResponseResponseBody));
}
function genUpscaleGenerateResponseResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenUpscaleGenerateResponseResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenUpscaleGenerateResponseResponseBody' from JSON`);
}
/** @internal */
exports.GenUpscaleGenerateResponseBody$inboundSchema = z.union([httperror_js_1.HTTPError$inboundSchema, studioapierror_js_1.StudioApiError$inboundSchema]);
/** @internal */
exports.GenUpscaleGenerateResponseBody$outboundSchema = z.union([httperror_js_1.HTTPError$outboundSchema, studioapierror_js_1.StudioApiError$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenUpscaleGenerateResponseBody$;
(function (GenUpscaleGenerateResponseBody$) {
    /** @deprecated use `GenUpscaleGenerateResponseBody$inboundSchema` instead. */
    GenUpscaleGenerateResponseBody$.inboundSchema = exports.GenUpscaleGenerateResponseBody$inboundSchema;
    /** @deprecated use `GenUpscaleGenerateResponseBody$outboundSchema` instead. */
    GenUpscaleGenerateResponseBody$.outboundSchema = exports.GenUpscaleGenerateResponseBody$outboundSchema;
})(GenUpscaleGenerateResponseBody$ || (exports.GenUpscaleGenerateResponseBody$ = GenUpscaleGenerateResponseBody$ = {}));
function genUpscaleGenerateResponseBodyToJSON(genUpscaleGenerateResponseBody) {
    return JSON.stringify(exports.GenUpscaleGenerateResponseBody$outboundSchema.parse(genUpscaleGenerateResponseBody));
}
function genUpscaleGenerateResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenUpscaleGenerateResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenUpscaleGenerateResponseBody' from JSON`);
}
/** @internal */
exports.GenUpscaleResponseBody$inboundSchema = z.union([httperror_js_1.HTTPError$inboundSchema, studioapierror_js_1.StudioApiError$inboundSchema]);
/** @internal */
exports.GenUpscaleResponseBody$outboundSchema = z.union([httperror_js_1.HTTPError$outboundSchema, studioapierror_js_1.StudioApiError$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GenUpscaleResponseBody$;
(function (GenUpscaleResponseBody$) {
    /** @deprecated use `GenUpscaleResponseBody$inboundSchema` instead. */
    GenUpscaleResponseBody$.inboundSchema = exports.GenUpscaleResponseBody$inboundSchema;
    /** @deprecated use `GenUpscaleResponseBody$outboundSchema` instead. */
    GenUpscaleResponseBody$.outboundSchema = exports.GenUpscaleResponseBody$outboundSchema;
})(GenUpscaleResponseBody$ || (exports.GenUpscaleResponseBody$ = GenUpscaleResponseBody$ = {}));
function genUpscaleResponseBodyToJSON(genUpscaleResponseBody) {
    return JSON.stringify(exports.GenUpscaleResponseBody$outboundSchema.parse(genUpscaleResponseBody));
}
function genUpscaleResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GenUpscaleResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GenUpscaleResponseBody' from JSON`);
}
