// livepeer/models/errors/studioapierror.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StudioApiError$ = exports.StudioApiError$outboundSchema = exports.StudioApiError$inboundSchema = exports.StudioApiError = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
class StudioApiError extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        if (err.errors != null)
            this.errors = err.errors;
        if (err.rawResponse != null)
            this.rawResponse = err.rawResponse;
        this.name = "StudioApiError";
    }
}
exports.StudioApiError = StudioApiError;
/** @internal */
exports.StudioApiError$inboundSchema = z.object({
    errors: z.array(z.string()).optional(),
    RawResponse: z.instanceof(Response).optional(),
})
    .transform((v) => {
    const remapped = (0, primitives_js_1.remap)(v, {
        "RawResponse": "rawResponse",
    });
    return new StudioApiError(remapped);
});
/** @internal */
exports.StudioApiError$outboundSchema = z.instanceof(StudioApiError)
    .transform(v => v.data$)
    .pipe(z.object({
    errors: z.array(z.string()).optional(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        rawResponse: "RawResponse",
    });
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StudioApiError$;
(function (StudioApiError$) {
    /** @deprecated use `StudioApiError$inboundSchema` instead. */
    StudioApiError$.inboundSchema = exports.StudioApiError$inboundSchema;
    /** @deprecated use `StudioApiError$outboundSchema` instead. */
    StudioApiError$.outboundSchema = exports.StudioApiError$outboundSchema;
})(StudioApiError$ || (exports.StudioApiError$ = StudioApiError$ = {}));
