// livepeer/models/components/bodygenimagetoimage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BodyGenImageToImage$ = exports.BodyGenImageToImage$outboundSchema = exports.BodyGenImageToImage$inboundSchema = exports.Image$ = exports.Image$outboundSchema = exports.Image$inboundSchema = void 0;
exports.imageToJSON = imageToJSON;
exports.imageFromJSON = imageFromJSON;
exports.bodyGenImageToImageToJSON = bodyGenImageToImageToJSON;
exports.bodyGenImageToImageFromJSON = bodyGenImageToImageFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const blobs_js_1 = require("../../types/blobs.js");
/** @internal */
exports.Image$inboundSchema = z
    .object({
    fileName: z.string(),
    content: z.union([
        z.instanceof((ReadableStream)),
        z.instanceof(Blob),
        z.instanceof(ArrayBuffer),
        z.instanceof(Uint8Array),
    ]),
});
/** @internal */
exports.Image$outboundSchema = z.object({
    fileName: z.string(),
    content: z.union([
        z.instanceof((ReadableStream)),
        z.instanceof(Blob),
        z.instanceof(ArrayBuffer),
        z.instanceof(Uint8Array),
    ]),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Image$;
(function (Image$) {
    /** @deprecated use `Image$inboundSchema` instead. */
    Image$.inboundSchema = exports.Image$inboundSchema;
    /** @deprecated use `Image$outboundSchema` instead. */
    Image$.outboundSchema = exports.Image$outboundSchema;
})(Image$ || (exports.Image$ = Image$ = {}));
function imageToJSON(image) {
    return JSON.stringify(exports.Image$outboundSchema.parse(image));
}
function imageFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Image$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Image' from JSON`);
}
/** @internal */
exports.BodyGenImageToImage$inboundSchema = z.object({
    image: z.lazy(() => exports.Image$inboundSchema),
    prompt: z.string(),
    guidance_scale: z.number().default(7.5),
    image_guidance_scale: z.number().default(1.5),
    loras: z.string().default(""),
    model_id: z.string().default("timbrooks/instruct-pix2pix"),
    negative_prompt: z.string().default(""),
    num_images_per_prompt: z.number().int().default(1),
    num_inference_steps: z.number().int().default(100),
    safety_check: z.boolean().default(true),
    seed: z.number().int().optional(),
    strength: z.number().default(0.8),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "guidance_scale": "guidanceScale",
        "image_guidance_scale": "imageGuidanceScale",
        "model_id": "modelId",
        "negative_prompt": "negativePrompt",
        "num_images_per_prompt": "numImagesPerPrompt",
        "num_inference_steps": "numInferenceSteps",
        "safety_check": "safetyCheck",
    });
});
/** @internal */
exports.BodyGenImageToImage$outboundSchema = z.object({
    image: z.lazy(() => exports.Image$outboundSchema).or(blobs_js_1.blobLikeSchema),
    prompt: z.string(),
    guidanceScale: z.number().default(7.5),
    imageGuidanceScale: z.number().default(1.5),
    loras: z.string().default(""),
    modelId: z.string().default("timbrooks/instruct-pix2pix"),
    negativePrompt: z.string().default(""),
    numImagesPerPrompt: z.number().int().default(1),
    numInferenceSteps: z.number().int().default(100),
    safetyCheck: z.boolean().default(true),
    seed: z.number().int().optional(),
    strength: z.number().default(0.8),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        guidanceScale: "guidance_scale",
        imageGuidanceScale: "image_guidance_scale",
        modelId: "model_id",
        negativePrompt: "negative_prompt",
        numImagesPerPrompt: "num_images_per_prompt",
        numInferenceSteps: "num_inference_steps",
        safetyCheck: "safety_check",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BodyGenImageToImage$;
(function (BodyGenImageToImage$) {
    /** @deprecated use `BodyGenImageToImage$inboundSchema` instead. */
    BodyGenImageToImage$.inboundSchema = exports.BodyGenImageToImage$inboundSchema;
    /** @deprecated use `BodyGenImageToImage$outboundSchema` instead. */
    BodyGenImageToImage$.outboundSchema = exports.BodyGenImageToImage$outboundSchema;
})(BodyGenImageToImage$ || (exports.BodyGenImageToImage$ = BodyGenImageToImage$ = {}));
function bodyGenImageToImageToJSON(bodyGenImageToImage) {
    return JSON.stringify(exports.BodyGenImageToImage$outboundSchema.parse(bodyGenImageToImage));
}
function bodyGenImageToImageFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BodyGenImageToImage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BodyGenImageToImage' from JSON`);
}
