// livepeer/models/components/clippayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClipPayload$ = exports.ClipPayload$outboundSchema = exports.ClipPayload$inboundSchema = void 0;
exports.clipPayloadToJSON = clipPayloadToJSON;
exports.clipPayloadFromJSON = clipPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ClipPayload$inboundSchema = z.object({
    playbackId: z.string(),
    startTime: z.number(),
    endTime: z.number().optional(),
    name: z.string().optional(),
    sessionId: z.string().optional(),
});
/** @internal */
exports.ClipPayload$outboundSchema = z.object({
    playbackId: z.string(),
    startTime: z.number(),
    endTime: z.number().optional(),
    name: z.string().optional(),
    sessionId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ClipPayload$;
(function (ClipPayload$) {
    /** @deprecated use `ClipPayload$inboundSchema` instead. */
    ClipPayload$.inboundSchema = exports.ClipPayload$inboundSchema;
    /** @deprecated use `ClipPayload$outboundSchema` instead. */
    ClipPayload$.outboundSchema = exports.ClipPayload$outboundSchema;
})(ClipPayload$ || (exports.ClipPayload$ = ClipPayload$ = {}));
function clipPayloadToJSON(clipPayload) {
    return JSON.stringify(exports.ClipPayload$outboundSchema.parse(clipPayload));
}
function clipPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ClipPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ClipPayload' from JSON`);
}
