// livepeer/models/components/ipfsfileinfo.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IpfsFileInfo$ = exports.IpfsFileInfo$outboundSchema = exports.IpfsFileInfo$inboundSchema = void 0;
exports.ipfsFileInfoToJSON = ipfsFileInfoToJSON;
exports.ipfsFileInfoFromJSON = ipfsFileInfoFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.IpfsFileInfo$inboundSchema = z.object({
    cid: z.string(),
    url: z.string().optional(),
    gatewayUrl: z.string().optional(),
});
/** @internal */
exports.IpfsFileInfo$outboundSchema = z.object({
    cid: z.string(),
    url: z.string().optional(),
    gatewayUrl: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IpfsFileInfo$;
(function (IpfsFileInfo$) {
    /** @deprecated use `IpfsFileInfo$inboundSchema` instead. */
    IpfsFileInfo$.inboundSchema = exports.IpfsFileInfo$inboundSchema;
    /** @deprecated use `IpfsFileInfo$outboundSchema` instead. */
    IpfsFileInfo$.outboundSchema = exports.IpfsFileInfo$outboundSchema;
})(IpfsFileInfo$ || (exports.IpfsFileInfo$ = IpfsFileInfo$ = {}));
function ipfsFileInfoToJSON(ipfsFileInfo) {
    return JSON.stringify(exports.IpfsFileInfo$outboundSchema.parse(ipfsFileInfo));
}
function ipfsFileInfoFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.IpfsFileInfo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'IpfsFileInfo' from JSON`);
}
