// livepeer/models/components/multistreamtarget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultistreamTarget$ = exports.MultistreamTarget$outboundSchema = exports.MultistreamTarget$inboundSchema = void 0;
exports.multistreamTargetToJSON = multistreamTargetToJSON;
exports.multistreamTargetFromJSON = multistreamTargetFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.MultistreamTarget$inboundSchema = z.object({
    id: z.string().optional(),
    name: z.string().optional(),
    userId: z.string().optional(),
    disabled: z.boolean().optional(),
    createdAt: z.number().optional(),
});
/** @internal */
exports.MultistreamTarget$outboundSchema = z.object({
    id: z.string().optional(),
    name: z.string().optional(),
    userId: z.string().optional(),
    disabled: z.boolean().optional(),
    createdAt: z.number().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MultistreamTarget$;
(function (MultistreamTarget$) {
    /** @deprecated use `MultistreamTarget$inboundSchema` instead. */
    MultistreamTarget$.inboundSchema = exports.MultistreamTarget$inboundSchema;
    /** @deprecated use `MultistreamTarget$outboundSchema` instead. */
    MultistreamTarget$.outboundSchema = exports.MultistreamTarget$outboundSchema;
})(MultistreamTarget$ || (exports.MultistreamTarget$ = MultistreamTarget$ = {}));
function multistreamTargetToJSON(multistreamTarget) {
    return JSON.stringify(exports.MultistreamTarget$outboundSchema.parse(multistreamTarget));
}
function multistreamTargetFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MultistreamTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MultistreamTarget' from JSON`);
}
