// livepeer/models/components/multistreamtargetpatchpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultistreamTargetPatchPayload$ = exports.MultistreamTargetPatchPayload$outboundSchema = exports.MultistreamTargetPatchPayload$inboundSchema = void 0;
exports.multistreamTargetPatchPayloadToJSON = multistreamTargetPatchPayloadToJSON;
exports.multistreamTargetPatchPayloadFromJSON = multistreamTargetPatchPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.MultistreamTargetPatchPayload$inboundSchema = z.object({
    name: z.string().optional(),
    url: z.string(),
    disabled: z.boolean().optional(),
});
/** @internal */
exports.MultistreamTargetPatchPayload$outboundSchema = z.object({
    name: z.string().optional(),
    url: z.string(),
    disabled: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MultistreamTargetPatchPayload$;
(function (MultistreamTargetPatchPayload$) {
    /** @deprecated use `MultistreamTargetPatchPayload$inboundSchema` instead. */
    MultistreamTargetPatchPayload$.inboundSchema = exports.MultistreamTargetPatchPayload$inboundSchema;
    /** @deprecated use `MultistreamTargetPatchPayload$outboundSchema` instead. */
    MultistreamTargetPatchPayload$.outboundSchema = exports.MultistreamTargetPatchPayload$outboundSchema;
})(MultistreamTargetPatchPayload$ || (exports.MultistreamTargetPatchPayload$ = MultistreamTargetPatchPayload$ = {}));
function multistreamTargetPatchPayloadToJSON(multistreamTargetPatchPayload) {
    return JSON.stringify(exports.MultistreamTargetPatchPayload$outboundSchema.parse(multistreamTargetPatchPayload));
}
function multistreamTargetPatchPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MultistreamTargetPatchPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MultistreamTargetPatchPayload' from JSON`);
}
