// livepeer/models/components/multistream.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Multistream$ = exports.Multistream$outboundSchema = exports.Multistream$inboundSchema = void 0;
exports.multistreamToJSON = multistreamToJSON;
exports.multistreamFromJSON = multistreamFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const target_js_1 = require("./target.js");
/** @internal */
exports.Multistream$inboundSchema = z.object({
    targets: z.array(target_js_1.Target$inboundSchema).optional(),
});
/** @internal */
exports.Multistream$outboundSchema = z.object({
    targets: z.array(target_js_1.Target$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Multistream$;
(function (Multistream$) {
    /** @deprecated use `Multistream$inboundSchema` instead. */
    Multistream$.inboundSchema = exports.Multistream$inboundSchema;
    /** @deprecated use `Multistream$outboundSchema` instead. */
    Multistream$.outboundSchema = exports.Multistream$outboundSchema;
})(Multistream$ || (exports.Multistream$ = Multistream$ = {}));
function multistreamToJSON(multistream) {
    return JSON.stringify(exports.Multistream$outboundSchema.parse(multistream));
}
function multistreamFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Multistream$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Multistream' from JSON`);
}
