// livepeer/models/components/newassetfromurlpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewAssetFromUrlPayload$ = exports.NewAssetFromUrlPayload$outboundSchema = exports.NewAssetFromUrlPayload$inboundSchema = void 0;
exports.newAssetFromUrlPayloadToJSON = newAssetFromUrlPayloadToJSON;
exports.newAssetFromUrlPayloadFromJSON = newAssetFromUrlPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const encryption_js_1 = require("./encryption.js");
const inputcreatorid_js_1 = require("./inputcreatorid.js");
const playbackpolicy_js_1 = require("./playbackpolicy.js");
const storage_js_1 = require("./storage.js");
const transcodeprofile_js_1 = require("./transcodeprofile.js");
/** @internal */
exports.NewAssetFromUrlPayload$inboundSchema = z.object({
    name: z.string(),
    staticMp4: z.boolean().optional(),
    playbackPolicy: z.nullable(playbackpolicy_js_1.PlaybackPolicy$inboundSchema).optional(),
    creatorId: inputcreatorid_js_1.InputCreatorId$inboundSchema.optional(),
    storage: storage_js_1.Storage$inboundSchema.optional(),
    url: z.string(),
    encryption: encryption_js_1.Encryption$inboundSchema.optional(),
    c2pa: z.boolean().optional(),
    profiles: z.nullable(z.array(transcodeprofile_js_1.TranscodeProfile$inboundSchema)).optional(),
    targetSegmentSizeSecs: z.number().optional(),
});
/** @internal */
exports.NewAssetFromUrlPayload$outboundSchema = z.object({
    name: z.string(),
    staticMp4: z.boolean().optional(),
    playbackPolicy: z.nullable(playbackpolicy_js_1.PlaybackPolicy$outboundSchema).optional(),
    creatorId: inputcreatorid_js_1.InputCreatorId$outboundSchema.optional(),
    storage: storage_js_1.Storage$outboundSchema.optional(),
    url: z.string(),
    encryption: encryption_js_1.Encryption$outboundSchema.optional(),
    c2pa: z.boolean().optional(),
    profiles: z.nullable(z.array(transcodeprofile_js_1.TranscodeProfile$outboundSchema)).optional(),
    targetSegmentSizeSecs: z.number().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NewAssetFromUrlPayload$;
(function (NewAssetFromUrlPayload$) {
    /** @deprecated use `NewAssetFromUrlPayload$inboundSchema` instead. */
    NewAssetFromUrlPayload$.inboundSchema = exports.NewAssetFromUrlPayload$inboundSchema;
    /** @deprecated use `NewAssetFromUrlPayload$outboundSchema` instead. */
    NewAssetFromUrlPayload$.outboundSchema = exports.NewAssetFromUrlPayload$outboundSchema;
})(NewAssetFromUrlPayload$ || (exports.NewAssetFromUrlPayload$ = NewAssetFromUrlPayload$ = {}));
function newAssetFromUrlPayloadToJSON(newAssetFromUrlPayload) {
    return JSON.stringify(exports.NewAssetFromUrlPayload$outboundSchema.parse(newAssetFromUrlPayload));
}
function newAssetFromUrlPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NewAssetFromUrlPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewAssetFromUrlPayload' from JSON`);
}
