// livepeer/models/components/newstreampayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewStreamPayload$ = exports.NewStreamPayload$outboundSchema = exports.NewStreamPayload$inboundSchema = exports.NewStreamPayloadRecordingSpec$ = exports.NewStreamPayloadRecordingSpec$outboundSchema = exports.NewStreamPayloadRecordingSpec$inboundSchema = void 0;
exports.newStreamPayloadRecordingSpecToJSON = newStreamPayloadRecordingSpecToJSON;
exports.newStreamPayloadRecordingSpecFromJSON = newStreamPayloadRecordingSpecFromJSON;
exports.newStreamPayloadToJSON = newStreamPayloadToJSON;
exports.newStreamPayloadFromJSON = newStreamPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const ffmpegprofile_js_1 = require("./ffmpegprofile.js");
const inputcreatorid_js_1 = require("./inputcreatorid.js");
const multistream_js_1 = require("./multistream.js");
const playbackpolicy_js_1 = require("./playbackpolicy.js");
const pull_js_1 = require("./pull.js");
const transcodeprofile_js_1 = require("./transcodeprofile.js");
const usertags_js_1 = require("./usertags.js");
/** @internal */
exports.NewStreamPayloadRecordingSpec$inboundSchema = z.object({
    profiles: z.nullable(z.array(transcodeprofile_js_1.TranscodeProfile$inboundSchema)).optional(),
});
/** @internal */
exports.NewStreamPayloadRecordingSpec$outboundSchema = z.object({
    profiles: z.nullable(z.array(transcodeprofile_js_1.TranscodeProfile$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NewStreamPayloadRecordingSpec$;
(function (NewStreamPayloadRecordingSpec$) {
    /** @deprecated use `NewStreamPayloadRecordingSpec$inboundSchema` instead. */
    NewStreamPayloadRecordingSpec$.inboundSchema = exports.NewStreamPayloadRecordingSpec$inboundSchema;
    /** @deprecated use `NewStreamPayloadRecordingSpec$outboundSchema` instead. */
    NewStreamPayloadRecordingSpec$.outboundSchema = exports.NewStreamPayloadRecordingSpec$outboundSchema;
})(NewStreamPayloadRecordingSpec$ || (exports.NewStreamPayloadRecordingSpec$ = NewStreamPayloadRecordingSpec$ = {}));
function newStreamPayloadRecordingSpecToJSON(newStreamPayloadRecordingSpec) {
    return JSON.stringify(exports.NewStreamPayloadRecordingSpec$outboundSchema.parse(newStreamPayloadRecordingSpec));
}
function newStreamPayloadRecordingSpecFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NewStreamPayloadRecordingSpec$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewStreamPayloadRecordingSpec' from JSON`);
}
/** @internal */
exports.NewStreamPayload$inboundSchema = z.object({
    name: z.string(),
    pull: pull_js_1.Pull$inboundSchema.optional(),
    creatorId: inputcreatorid_js_1.InputCreatorId$inboundSchema.optional(),
    playbackPolicy: z.nullable(playbackpolicy_js_1.PlaybackPolicy$inboundSchema).optional(),
    profiles: z.nullable(z.array(ffmpegprofile_js_1.FfmpegProfile$inboundSchema)).optional(),
    record: z.boolean().optional(),
    recordingSpec: z.lazy(() => exports.NewStreamPayloadRecordingSpec$inboundSchema)
        .optional(),
    multistream: multistream_js_1.Multistream$inboundSchema.optional(),
    userTags: z.record(usertags_js_1.UserTags$inboundSchema).optional(),
});
/** @internal */
exports.NewStreamPayload$outboundSchema = z.object({
    name: z.string(),
    pull: pull_js_1.Pull$outboundSchema.optional(),
    creatorId: inputcreatorid_js_1.InputCreatorId$outboundSchema.optional(),
    playbackPolicy: z.nullable(playbackpolicy_js_1.PlaybackPolicy$outboundSchema).optional(),
    profiles: z.nullable(z.array(ffmpegprofile_js_1.FfmpegProfile$outboundSchema)).optional(),
    record: z.boolean().optional(),
    recordingSpec: z.lazy(() => exports.NewStreamPayloadRecordingSpec$outboundSchema)
        .optional(),
    multistream: multistream_js_1.Multistream$outboundSchema.optional(),
    userTags: z.record(usertags_js_1.UserTags$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NewStreamPayload$;
(function (NewStreamPayload$) {
    /** @deprecated use `NewStreamPayload$inboundSchema` instead. */
    NewStreamPayload$.inboundSchema = exports.NewStreamPayload$inboundSchema;
    /** @deprecated use `NewStreamPayload$outboundSchema` instead. */
    NewStreamPayload$.outboundSchema = exports.NewStreamPayload$outboundSchema;
})(NewStreamPayload$ || (exports.NewStreamPayload$ = NewStreamPayload$ = {}));
function newStreamPayloadToJSON(newStreamPayload) {
    return JSON.stringify(exports.NewStreamPayload$outboundSchema.parse(newStreamPayload));
}
function newStreamPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NewStreamPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewStreamPayload' from JSON`);
}
