// livepeer/models/components/room.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Room$ = exports.Room$outboundSchema = exports.Room$inboundSchema = exports.Participants$ = exports.Participants$outboundSchema = exports.Participants$inboundSchema = void 0;
exports.participantsToJSON = participantsToJSON;
exports.participantsFromJSON = participantsFromJSON;
exports.roomToJSON = roomToJSON;
exports.roomFromJSON = roomFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.Participants$inboundSchema = z.object({
    identity: z.string().optional(),
    name: z.string().optional(),
    joinedAt: z.number().int().optional(),
    leftAt: z.number().int().optional(),
});
/** @internal */
exports.Participants$outboundSchema = z.object({
    identity: z.string().optional(),
    name: z.string().optional(),
    joinedAt: z.number().int().optional(),
    leftAt: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Participants$;
(function (Participants$) {
    /** @deprecated use `Participants$inboundSchema` instead. */
    Participants$.inboundSchema = exports.Participants$inboundSchema;
    /** @deprecated use `Participants$outboundSchema` instead. */
    Participants$.outboundSchema = exports.Participants$outboundSchema;
})(Participants$ || (exports.Participants$ = Participants$ = {}));
function participantsToJSON(participants) {
    return JSON.stringify(exports.Participants$outboundSchema.parse(participants));
}
function participantsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Participants$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Participants' from JSON`);
}
/** @internal */
exports.Room$inboundSchema = z
    .object({
    id: z.string(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    egressId: z.string().optional(),
    participants: z.record(z.lazy(() => exports.Participants$inboundSchema)),
});
/** @internal */
exports.Room$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    egressId: z.string().optional(),
    participants: z.record(z.lazy(() => exports.Participants$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Room$;
(function (Room$) {
    /** @deprecated use `Room$inboundSchema` instead. */
    Room$.inboundSchema = exports.Room$inboundSchema;
    /** @deprecated use `Room$outboundSchema` instead. */
    Room$.outboundSchema = exports.Room$outboundSchema;
})(Room$ || (exports.Room$ = Room$ = {}));
function roomToJSON(room) {
    return JSON.stringify(exports.Room$outboundSchema.parse(room));
}
function roomFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Room$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Room' from JSON`);
}
