// livepeer/models/components/signingkey.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SigningKey$ = exports.SigningKey$outboundSchema = exports.SigningKey$inboundSchema = void 0;
exports.signingKeyToJSON = signingKeyToJSON;
exports.signingKeyFromJSON = signingKeyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.SigningKey$inboundSchema = z.object({
    id: z.string().optional(),
    name: z.string().optional(),
    userId: z.string().optional(),
    createdAt: z.number().optional(),
    lastSeen: z.number().optional(),
    publicKey: z.string(),
    disabled: z.boolean().optional(),
    projectId: z.string().optional(),
});
/** @internal */
exports.SigningKey$outboundSchema = z.object({
    id: z.string().optional(),
    name: z.string().optional(),
    userId: z.string().optional(),
    createdAt: z.number().optional(),
    lastSeen: z.number().optional(),
    publicKey: z.string(),
    disabled: z.boolean().optional(),
    projectId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SigningKey$;
(function (SigningKey$) {
    /** @deprecated use `SigningKey$inboundSchema` instead. */
    SigningKey$.inboundSchema = exports.SigningKey$inboundSchema;
    /** @deprecated use `SigningKey$outboundSchema` instead. */
    SigningKey$.outboundSchema = exports.SigningKey$outboundSchema;
})(SigningKey$ || (exports.SigningKey$ = SigningKey$ = {}));
function signingKeyToJSON(signingKey) {
    return JSON.stringify(exports.SigningKey$outboundSchema.parse(signingKey));
}
function signingKeyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SigningKey$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SigningKey' from JSON`);
}
