// livepeer/models/components/streampatchpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamPatchPayload$ = exports.StreamPatchPayload$outboundSchema = exports.StreamPatchPayload$inboundSchema = void 0;
exports.streamPatchPayloadToJSON = streamPatchPayloadToJSON;
exports.streamPatchPayloadFromJSON = streamPatchPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const ffmpegprofile_js_1 = require("./ffmpegprofile.js");
const inputcreatorid_js_1 = require("./inputcreatorid.js");
const multistream_js_1 = require("./multistream.js");
const playbackpolicy_js_1 = require("./playbackpolicy.js");
const recordingspec_js_1 = require("./recordingspec.js");
const usertags_js_1 = require("./usertags.js");
/** @internal */
exports.StreamPatchPayload$inboundSchema = z.object({
    creatorId: inputcreatorid_js_1.InputCreatorId$inboundSchema.optional(),
    record: z.boolean().optional(),
    suspended: z.boolean().optional(),
    multistream: multistream_js_1.Multistream$inboundSchema.optional(),
    playbackPolicy: z.nullable(playbackpolicy_js_1.PlaybackPolicy$inboundSchema).optional(),
    profiles: z.nullable(z.array(ffmpegprofile_js_1.FfmpegProfile$inboundSchema)).optional(),
    recordingSpec: recordingspec_js_1.RecordingSpec$inboundSchema.optional(),
    userTags: z.record(usertags_js_1.UserTags$inboundSchema).optional(),
    name: z.string().optional(),
});
/** @internal */
exports.StreamPatchPayload$outboundSchema = z.object({
    creatorId: inputcreatorid_js_1.InputCreatorId$outboundSchema.optional(),
    record: z.boolean().optional(),
    suspended: z.boolean().optional(),
    multistream: multistream_js_1.Multistream$outboundSchema.optional(),
    playbackPolicy: z.nullable(playbackpolicy_js_1.PlaybackPolicy$outboundSchema).optional(),
    profiles: z.nullable(z.array(ffmpegprofile_js_1.FfmpegProfile$outboundSchema)).optional(),
    recordingSpec: recordingspec_js_1.RecordingSpec$outboundSchema.optional(),
    userTags: z.record(usertags_js_1.UserTags$outboundSchema).optional(),
    name: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StreamPatchPayload$;
(function (StreamPatchPayload$) {
    /** @deprecated use `StreamPatchPayload$inboundSchema` instead. */
    StreamPatchPayload$.inboundSchema = exports.StreamPatchPayload$inboundSchema;
    /** @deprecated use `StreamPatchPayload$outboundSchema` instead. */
    StreamPatchPayload$.outboundSchema = exports.StreamPatchPayload$outboundSchema;
})(StreamPatchPayload$ || (exports.StreamPatchPayload$ = StreamPatchPayload$ = {}));
function streamPatchPayloadToJSON(streamPatchPayload) {
    return JSON.stringify(exports.StreamPatchPayload$outboundSchema.parse(streamPatchPayload));
}
function streamPatchPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StreamPatchPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamPatchPayload' from JSON`);
}
