// livepeer/models/components/studioapierror.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StudioApiError$ = exports.StudioApiError$outboundSchema = exports.StudioApiError$inboundSchema = void 0;
exports.studioApiErrorToJSON = studioApiErrorToJSON;
exports.studioApiErrorFromJSON = studioApiErrorFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.StudioApiError$inboundSchema = z.object({
    errors: z.array(z.string()).optional(),
    RawResponse: z.instanceof(Response).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.StudioApiError$outboundSchema = z.object({
    errors: z.array(z.string()).optional(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StudioApiError$;
(function (StudioApiError$) {
    /** @deprecated use `StudioApiError$inboundSchema` instead. */
    StudioApiError$.inboundSchema = exports.StudioApiError$inboundSchema;
    /** @deprecated use `StudioApiError$outboundSchema` instead. */
    StudioApiError$.outboundSchema = exports.StudioApiError$outboundSchema;
})(StudioApiError$ || (exports.StudioApiError$ = StudioApiError$ = {}));
function studioApiErrorToJSON(studioApiError) {
    return JSON.stringify(exports.StudioApiError$outboundSchema.parse(studioApiError));
}
function studioApiErrorFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.StudioApiError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StudioApiError' from JSON`);
}
