// livepeer/models/components/target.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Target$ = exports.Target$outboundSchema = exports.Target$inboundSchema = exports.TargetSpec$ = exports.TargetSpec$outboundSchema = exports.TargetSpec$inboundSchema = void 0;
exports.targetSpecToJSON = targetSpecToJSON;
exports.targetSpecFromJSON = targetSpecFromJSON;
exports.targetToJSON = targetToJSON;
exports.targetFromJSON = targetFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.TargetSpec$inboundSchema = z.object({
    name: z.string().optional(),
    url: z.string(),
});
/** @internal */
exports.TargetSpec$outboundSchema = z.object({
    name: z.string().optional(),
    url: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TargetSpec$;
(function (TargetSpec$) {
    /** @deprecated use `TargetSpec$inboundSchema` instead. */
    TargetSpec$.inboundSchema = exports.TargetSpec$inboundSchema;
    /** @deprecated use `TargetSpec$outboundSchema` instead. */
    TargetSpec$.outboundSchema = exports.TargetSpec$outboundSchema;
})(TargetSpec$ || (exports.TargetSpec$ = TargetSpec$ = {}));
function targetSpecToJSON(targetSpec) {
    return JSON.stringify(exports.TargetSpec$outboundSchema.parse(targetSpec));
}
function targetSpecFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TargetSpec$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TargetSpec' from JSON`);
}
/** @internal */
exports.Target$inboundSchema = z
    .object({
    profile: z.string(),
    videoOnly: z.boolean().default(false),
    id: z.string().optional(),
    spec: z.lazy(() => exports.TargetSpec$inboundSchema).optional(),
});
/** @internal */
exports.Target$outboundSchema = z.object({
    profile: z.string(),
    videoOnly: z.boolean().default(false),
    id: z.string().optional(),
    spec: z.lazy(() => exports.TargetSpec$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Target$;
(function (Target$) {
    /** @deprecated use `Target$inboundSchema` instead. */
    Target$.inboundSchema = exports.Target$inboundSchema;
    /** @deprecated use `Target$outboundSchema` instead. */
    Target$.outboundSchema = exports.Target$outboundSchema;
})(Target$ || (exports.Target$ = Target$ = {}));
function targetToJSON(target) {
    return JSON.stringify(exports.Target$outboundSchema.parse(target));
}
function targetFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Target$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Target' from JSON`);
}
