// livepeer/models/components/textresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextResponse$ = exports.TextResponse$outboundSchema = exports.TextResponse$inboundSchema = void 0;
exports.textResponseToJSON = textResponseToJSON;
exports.textResponseFromJSON = textResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const chunk_js_1 = require("./chunk.js");
/** @internal */
exports.TextResponse$inboundSchema = z.object({
    text: z.string(),
    chunks: z.array(chunk_js_1.Chunk$inboundSchema),
});
/** @internal */
exports.TextResponse$outboundSchema = z.object({
    text: z.string(),
    chunks: z.array(chunk_js_1.Chunk$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TextResponse$;
(function (TextResponse$) {
    /** @deprecated use `TextResponse$inboundSchema` instead. */
    TextResponse$.inboundSchema = exports.TextResponse$inboundSchema;
    /** @deprecated use `TextResponse$outboundSchema` instead. */
    TextResponse$.outboundSchema = exports.TextResponse$outboundSchema;
})(TextResponse$ || (exports.TextResponse$ = TextResponse$ = {}));
function textResponseToJSON(textResponse) {
    return JSON.stringify(exports.TextResponse$outboundSchema.parse(textResponse));
}
function textResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TextResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TextResponse' from JSON`);
}
