// livepeer/models/components/usagemetric.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageMetric$ = exports.UsageMetric$outboundSchema = exports.UsageMetric$inboundSchema = void 0;
exports.usageMetricToJSON = usageMetricToJSON;
exports.usageMetricFromJSON = usageMetricFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.UsageMetric$inboundSchema = z.object({
    UserID: z.string().optional(),
    CreatorID: z.string().optional(),
    DeliveryUsageMins: z.number().optional(),
    TotalUsageMins: z.number().optional(),
    StorageUsageMins: z.number().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "UserID": "userID",
        "CreatorID": "creatorID",
        "DeliveryUsageMins": "deliveryUsageMins",
        "TotalUsageMins": "totalUsageMins",
        "StorageUsageMins": "storageUsageMins",
    });
});
/** @internal */
exports.UsageMetric$outboundSchema = z.object({
    userID: z.string().optional(),
    creatorID: z.string().optional(),
    deliveryUsageMins: z.number().optional(),
    totalUsageMins: z.number().optional(),
    storageUsageMins: z.number().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userID: "UserID",
        creatorID: "CreatorID",
        deliveryUsageMins: "DeliveryUsageMins",
        totalUsageMins: "TotalUsageMins",
        storageUsageMins: "StorageUsageMins",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsageMetric$;
(function (UsageMetric$) {
    /** @deprecated use `UsageMetric$inboundSchema` instead. */
    UsageMetric$.inboundSchema = exports.UsageMetric$inboundSchema;
    /** @deprecated use `UsageMetric$outboundSchema` instead. */
    UsageMetric$.outboundSchema = exports.UsageMetric$outboundSchema;
})(UsageMetric$ || (exports.UsageMetric$ = UsageMetric$ = {}));
function usageMetricToJSON(usageMetric) {
    return JSON.stringify(exports.UsageMetric$outboundSchema.parse(usageMetric));
}
function usageMetricFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UsageMetric$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UsageMetric' from JSON`);
}
