// livepeer/models/components/videoresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoResponse$ = exports.VideoResponse$outboundSchema = exports.VideoResponse$inboundSchema = void 0;
exports.videoResponseToJSON = videoResponseToJSON;
exports.videoResponseFromJSON = videoResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const media_js_1 = require("./media.js");
/** @internal */
exports.VideoResponse$inboundSchema = z.object({
    images: z.array(media_js_1.Media$inboundSchema),
});
/** @internal */
exports.VideoResponse$outboundSchema = z.object({
    images: z.array(media_js_1.Media$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var VideoResponse$;
(function (VideoResponse$) {
    /** @deprecated use `VideoResponse$inboundSchema` instead. */
    VideoResponse$.inboundSchema = exports.VideoResponse$inboundSchema;
    /** @deprecated use `VideoResponse$outboundSchema` instead. */
    VideoResponse$.outboundSchema = exports.VideoResponse$outboundSchema;
})(VideoResponse$ || (exports.VideoResponse$ = VideoResponse$ = {}));
function videoResponseToJSON(videoResponse) {
    return JSON.stringify(exports.VideoResponse$outboundSchema.parse(videoResponse));
}
function videoResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.VideoResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'VideoResponse' from JSON`);
}
