// livepeer/models/components/viewershipmetric.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewershipMetric$ = exports.ViewershipMetric$outboundSchema = exports.ViewershipMetric$inboundSchema = void 0;
exports.viewershipMetricToJSON = viewershipMetricToJSON;
exports.viewershipMetricFromJSON = viewershipMetricFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ViewershipMetric$inboundSchema = z.object({
    playbackId: z.string().optional(),
    creatorId: z.string().optional(),
    viewerId: z.string().optional(),
    dStorageUrl: z.string().optional(),
    timestamp: z.number().optional(),
    device: z.string().optional(),
    deviceType: z.string().optional(),
    cpu: z.string().optional(),
    os: z.string().optional(),
    browser: z.string().optional(),
    browserEngine: z.string().optional(),
    continent: z.string().optional(),
    country: z.string().optional(),
    subdivision: z.string().optional(),
    timezone: z.string().optional(),
    geohash: z.string().optional(),
    viewCount: z.number().int(),
    playtimeMins: z.number(),
    ttffMs: z.number().optional(),
    rebufferRatio: z.number().optional(),
    errorRate: z.number().optional(),
    exitsBeforeStart: z.number().optional(),
});
/** @internal */
exports.ViewershipMetric$outboundSchema = z.object({
    playbackId: z.string().optional(),
    creatorId: z.string().optional(),
    viewerId: z.string().optional(),
    dStorageUrl: z.string().optional(),
    timestamp: z.number().optional(),
    device: z.string().optional(),
    deviceType: z.string().optional(),
    cpu: z.string().optional(),
    os: z.string().optional(),
    browser: z.string().optional(),
    browserEngine: z.string().optional(),
    continent: z.string().optional(),
    country: z.string().optional(),
    subdivision: z.string().optional(),
    timezone: z.string().optional(),
    geohash: z.string().optional(),
    viewCount: z.number().int(),
    playtimeMins: z.number(),
    ttffMs: z.number().optional(),
    rebufferRatio: z.number().optional(),
    errorRate: z.number().optional(),
    exitsBeforeStart: z.number().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ViewershipMetric$;
(function (ViewershipMetric$) {
    /** @deprecated use `ViewershipMetric$inboundSchema` instead. */
    ViewershipMetric$.inboundSchema = exports.ViewershipMetric$inboundSchema;
    /** @deprecated use `ViewershipMetric$outboundSchema` instead. */
    ViewershipMetric$.outboundSchema = exports.ViewershipMetric$outboundSchema;
})(ViewershipMetric$ || (exports.ViewershipMetric$ = ViewershipMetric$ = {}));
function viewershipMetricToJSON(viewershipMetric) {
    return JSON.stringify(exports.ViewershipMetric$outboundSchema.parse(viewershipMetric));
}
function viewershipMetricFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ViewershipMetric$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ViewershipMetric' from JSON`);
}
