// livepeer/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Livepeer = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const accesscontrol_js_1 = require("./accesscontrol.js");
const asset_js_1 = require("./asset.js");
const generate_js_1 = require("./generate.js");
const metrics_js_1 = require("./metrics.js");
const multistream_js_1 = require("./multistream.js");
const playback_js_1 = require("./playback.js");
const room_js_1 = require("./room.js");
const session_js_1 = require("./session.js");
const stream_js_1 = require("./stream.js");
const task_js_1 = require("./task.js");
const transcode_js_1 = require("./transcode.js");
const webhook_js_1 = require("./webhook.js");
class Livepeer extends sdks_js_1.ClientSDK {
    get stream() {
        var _a;
        return ((_a = this._stream) !== null && _a !== void 0 ? _a : (this._stream = new stream_js_1.Stream(this._options)));
    }
    get multistream() {
        var _a;
        return ((_a = this._multistream) !== null && _a !== void 0 ? _a : (this._multistream = new multistream_js_1.Multistream(this._options)));
    }
    get webhook() {
        var _a;
        return ((_a = this._webhook) !== null && _a !== void 0 ? _a : (this._webhook = new webhook_js_1.Webhook(this._options)));
    }
    get asset() {
        var _a;
        return ((_a = this._asset) !== null && _a !== void 0 ? _a : (this._asset = new asset_js_1.Asset(this._options)));
    }
    get session() {
        var _a;
        return ((_a = this._session) !== null && _a !== void 0 ? _a : (this._session = new session_js_1.Session(this._options)));
    }
    get room() {
        var _a;
        return ((_a = this._room) !== null && _a !== void 0 ? _a : (this._room = new room_js_1.Room(this._options)));
    }
    get metrics() {
        var _a;
        return ((_a = this._metrics) !== null && _a !== void 0 ? _a : (this._metrics = new metrics_js_1.Metrics(this._options)));
    }
    get accessControl() {
        var _a;
        return ((_a = this._accessControl) !== null && _a !== void 0 ? _a : (this._accessControl = new accesscontrol_js_1.AccessControl(this._options)));
    }
    get task() {
        var _a;
        return ((_a = this._task) !== null && _a !== void 0 ? _a : (this._task = new task_js_1.Task(this._options)));
    }
    get transcode() {
        var _a;
        return ((_a = this._transcode) !== null && _a !== void 0 ? _a : (this._transcode = new transcode_js_1.Transcode(this._options)));
    }
    get playback() {
        var _a;
        return ((_a = this._playback) !== null && _a !== void 0 ? _a : (this._playback = new playback_js_1.Playback(this._options)));
    }
    get generate() {
        var _a;
        return ((_a = this._generate) !== null && _a !== void 0 ? _a : (this._generate = new generate_js_1.Generate(this._options)));
    }
}
exports.Livepeer = Livepeer;
